{"cells":[{"cell_type":"code","execution_count":23,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":11661,"status":"ok","timestamp":1664477425137,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"A3ovee7A__RC","outputId":"77c39c65-a810-416d-df11-08972a2f66f7"},"outputs":[{"data":{"text/plain":["2712"]},"execution_count":23,"metadata":{},"output_type":"execute_result"}],"source":["import pandas as pd\n","import numpy as np \n","import gc\n","import seaborn as sns \n","import matplotlib.pyplot as plt\n","from sklearn.manifold import TSNE\n","from sklearn.decomposition import LatentDirichletAllocation\n","from sklearn.feature_extraction.text import CountVectorizer\n","from sklearn.cluster import MiniBatchKMeans, KMeans\n","from sklearn.feature_extraction.text import TfidfVectorizer # Why can't this be integrated with the previous cell\n","import nltk \n","import string\n","import re \n","from nltk.corpus import stopwords\n","import bokeh\n","from bokeh.models import ColumnDataSource, HoverTool, LinearColorMapper, CustomJS, Slider, TapTool, TextInput\n","from bokeh.transform import linear_cmap, transform\n","from bokeh.io import output_file, show, output_notebook\n","from bokeh.plotting import figure\n","from bokeh.palettes import Category20\n","from bokeh.layouts import layout\n","\n","\n","data=pd.read_csv('midas-data.csv')\n","gc.collect()"]},{"cell_type":"code","execution_count":24,"metadata":{"executionInfo":{"elapsed":5292,"status":"ok","timestamp":1664477466953,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"54nmEISyvTQH"},"outputs":[],"source":["data.dropna(inplace=True)\n","def preprocess_text(text: str, remove_stopwords: bool) -> str:\n","    text = re.sub(r\"http\\S+\", \"\", text)\n","    # remove special chars and numbers\n","    text = re.sub(\"[^A-Za-z]+\", \" \", text)\n","    # remove stopwords\n","    if remove_stopwords:\n","        tokens = nltk.word_tokenize(text)\n","        tokens = [w for w in tokens if not w.lower() in stopwords.words(\"english\")]\n","        text = text.lower().strip()\n","        text = \" \".join(tokens)\n","    return text\n","\n","data['cleaned'] = data['abstract'].apply(lambda x: preprocess_text(x, remove_stopwords=True))\n","data['cleaned']=data['cleaned'].str.lower()"]},{"cell_type":"markdown","metadata":{"id":"Bvhkhm_HY3Z8"},"source":["Now, that the abstract has been pre processed, it is time to  convert it into a format \n","that can be handled by our algorithms. For this purpose, we will be using tf-idf.  The tf-idf which stands for term frequency - invrse document frequency, converts the corpus to a numerical format \n","by bringing out specific terms, weighing very rare or  very common terms differently in order to assign them a low score. "]},{"cell_type":"code","execution_count":25,"metadata":{"executionInfo":{"elapsed":192,"status":"ok","timestamp":1664477473374,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"4aKtbbpvaBVO"},"outputs":[],"source":["def vectorize(text):\n","    \n","    vectorizer = TfidfVectorizer()\n","    X = vectorizer.fit_transform(text)\n","    return X"]},{"cell_type":"code","execution_count":27,"metadata":{"executionInfo":{"elapsed":409,"status":"ok","timestamp":1664477476814,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"UAUT5nqNaFno"},"outputs":[],"source":["text = data['cleaned'].values\n","\n","X = vectorize(text)"]},{"cell_type":"markdown","metadata":{"id":"M_0F8Rr7c8A8"},"source":[" We will apply the t-distributed stochastic neighbourhood embedding (t-SNE) to reduce the complexity of our dimensionality reduction step and filter noise.  "]},{"cell_type":"code","execution_count":48,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":13514,"status":"ok","timestamp":1664477493604,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"op0tr7ymgjpB","outputId":"61c45acb-8d47-4cdf-e62d-f19193cc4e39"},"outputs":[{"name":"stdout","output_type":"stream","text":["[t-SNE] Computing 151 nearest neighbors...\n","[t-SNE] Indexed 398 samples in 0.024s...\n","[t-SNE] Computed neighbors for 398 samples in 0.757s...\n","[t-SNE] Computed conditional probabilities for sample 398 / 398\n","[t-SNE] Mean sigma: 0.701574\n"]},{"name":"stderr","output_type":"stream","text":["/opt/anaconda3/lib/python3.7/site-packages/sklearn/manifold/t_sne.py:344: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  error = np.finfo(np.float).max\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/manifold/t_sne.py:345: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  best_error = np.finfo(np.float).max\n"]},{"name":"stdout","output_type":"stream","text":["[t-SNE] KL divergence after 250 iterations with early exaggeration: 66.829857\n"]},{"name":"stderr","output_type":"stream","text":["/opt/anaconda3/lib/python3.7/site-packages/sklearn/manifold/t_sne.py:344: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  error = np.finfo(np.float).max\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/manifold/t_sne.py:345: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  best_error = np.finfo(np.float).max\n"]},{"name":"stdout","output_type":"stream","text":["[t-SNE] KL divergence after 1000 iterations: 0.471719\n"]}],"source":["tsne = TSNE(verbose=1, perplexity=50)  \n","X_embedded = tsne.fit_transform(X.toarray())"]},{"cell_type":"markdown","metadata":{"id":"BSynIMbQgYI1"},"source":["K means will be run on the vectorized text. Given the number of clusters, k, k-means will categorize each vector by taking the mean distance to a randomly intalized centroid. The centroids are updated iteratively. \n","\n","To find the best value for k-means we will look athe disortion at different k values. \n","\n","When we plot disortion against k, there will be a k value after which decreases in disortion are minimal. These are the desired clusters."]},{"cell_type":"code","execution_count":32,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":1631,"status":"ok","timestamp":1664477499710,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"qL_SFUbonkIS","outputId":"1b8221ab-1eb5-475d-ff17-4cb4abb9d9b1"},"outputs":[{"name":"stderr","output_type":"stream","text":["/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n"]},{"name":"stdout","output_type":"stream","text":["Found distortion for 2 clusters\n","[8.413246676804242]\n","Found distortion for 3 clusters\n","[8.413246676804242, 7.074224692827318]\n"]},{"name":"stderr","output_type":"stream","text":["/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n"]},{"name":"stdout","output_type":"stream","text":["Found distortion for 4 clusters\n","[8.413246676804242, 7.074224692827318, 6.207844998471816]\n","Found distortion for 5 clusters\n","[8.413246676804242, 7.074224692827318, 6.207844998471816, 5.6291296787750875]\n"]},{"name":"stderr","output_type":"stream","text":["/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n"]},{"name":"stdout","output_type":"stream","text":["Found distortion for 6 clusters\n","[8.413246676804242, 7.074224692827318, 6.207844998471816, 5.6291296787750875, 5.095171122178999]\n","Found distortion for 7 clusters\n","[8.413246676804242, 7.074224692827318, 6.207844998471816, 5.6291296787750875, 5.095171122178999, 4.756172518517556]\n"]},{"name":"stderr","output_type":"stream","text":["/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n"]},{"name":"stdout","output_type":"stream","text":["Found distortion for 8 clusters\n","[8.413246676804242, 7.074224692827318, 6.207844998471816, 5.6291296787750875, 5.095171122178999, 4.756172518517556, 4.431766920608667]\n","Found distortion for 9 clusters\n","[8.413246676804242, 7.074224692827318, 6.207844998471816, 5.6291296787750875, 5.095171122178999, 4.756172518517556, 4.431766920608667, 4.191264869496754]\n"]},{"name":"stderr","output_type":"stream","text":["/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n"]}],"source":["from sklearn import metrics\n","from scipy.spatial.distance import cdist\n","\n","# run kmeans with many different k\n","distortions = []\n","K = range(2, 10)\n","for k in K:\n","    k_means = KMeans(n_clusters=k, random_state=42).fit(X_embedded)\n","    k_means.fit(X_embedded)\n","    distortions.append(sum(np.min(cdist(X_embedded, k_means.cluster_centers_, 'euclidean'), axis=1)) / X.shape[0])\n","    print('Found distortion for {} clusters'.format(k))\n","    print(distortions)"]},{"cell_type":"code","execution_count":33,"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":295},"executionInfo":{"elapsed":566,"status":"ok","timestamp":1664477514100,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"2JkuJoyBYDsj","outputId":"a04cb268-afa4-4829-c26d-a0ebde2bef65"},"outputs":[{"data":{"image/png":"iVBORw0KGgoAAAANSUhEUgAAAXgAAAEWCAYAAABsY4yMAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3deZyW8/7H8denppSKSNEqWbKEopyEROknR8JxDlHZzsl2IrL8LNmPg0NldyQiZEl2jr2yt+GI5NAiKcZpU0mpz++P7zW/xpi5557pvua673vez8fjfsw9933PfX3ue2bec83n+l7fr7k7IiKSf2okXYCIiMRDAS8ikqcU8CIieUoBLyKSpxTwIiJ5SgEvIpKnFPAxMbMrzeyhKthOazNzMyuIPp9gZn+Oe7tVIZOvxcxGm9m1lfg6N7MdMlFDGc9/gJnNiuv5S9lerK+nsszsEjO7N6bnnmtmPcq4r1I/F7lCAV9JZrai2GW9mf1U7PMTMryt0Wa2psQ2P87kNiqr2B+Y6SVu3yqqeW6az1MlfxCzjbu/5e5t43jubP1jb2bdzOyb4re5+3XunnW15joFfCW5e/2iC/A10LvYbQ/HsMkbi2/T3feMYRsbo56ZtSv2+fHAnKSKEREFfNxqm9mDZvajmX1qZh2L7jCzZmb2pJkVmtkcMzs7g9vd3swmm9kyM3vGzLYstt0jolqWRnt4u0S3n2xmzxV73Jdm9nixz+ebWfsU2xwDnFjs8wHAg8UfUNZrNrNDgUuAY0v572RbM3sneg9fMbOtynst0X0dzGx69HWPAXXKKtzMdjCzidH79UP0+OJ6mNl/zGyJmd1hZhZ9XQ0zu8zM5pnZ99H3evPovgfMbEh0vXn0X86Zxba32IJf7c1G7YTzzezfUT2PmVmdYvdfaGYLzexbM/tzWS0XM/sbcABwe/Se3l7e64m+7hQzmxnd97KZbZvifUv1/s81s4vN7LPoue43szpmVg94CWhmG/4bbVb8Pzjb8F/hydHP3RIzO93MOkXvy9Lir8fMtjezN8zsv9H372Eza1hW3SleTwMze9PMbi3+nuQ0d9dlIy/AXKBHiduuBFYDhwE1gb8D70f31QCmAZcDtYE2wGzgf8p4/tHAtWXc1xpwoCD6fAKwAGgH1AOeBB6K7tsJWAkcAtQCLgS+LFbD0qi2psA8YEH0dW2AJUCNFNtvDcyPXusuwCygBzA3ndccvV8PlXjuCcBXUd11o8+vT+O11I7qPze67xhgbYr3cCxwaVRjHWD/Yvc58DzQEGgFFAKHRvedEm2zDVAfGA+MKXbfc9H146PX8Vix+56JrncDvinxszQZaAZsCcwETo/uOxRYBOwGbEr4o+rADmW8rgnAn0vclur1HBm9nl2AAuAy4N0ynrvM97/Y65gBtIxexztF73/J11zy+8+Gn6m7o+9HT8Lv0tNAE6A58D1wYPT4HaI6NgEaA5OAEal+P0v+bgGNove91J+RXL1oDz5eb7v7i+6+jvDLWNRW6QQ0dver3X2Nu88GRgLHpXiu86M9l6LLAykeO8bdZ7j7SmAo8CczqwkcC7zg7q+6+1rgJkJwdolq+BFoDxwIvAwsMLOdo8/fcvf1Kbb5DRtC/URK7L1X8jUD3O/uX7j7T8DjUX2kei1AZ0LojHD3te4+DpiSYhtrgW2BZu6+2t3fLnH/9e6+1N2/Bt4sVsMJwDB3n+3uK4CLgeMsHPCeCBxgZjWArsCNwH7R1x0Y3V+WW939W3dfDDxXbHt/it6PT919FXBViudIpazXcxrwd3ef6e6/ANcB7cvYi0/1/he53d3nR6/jb0DfCtZ5TfT9eIXwx2Ssu3/v7guAt4AOAO7+ZVTHz+5eCAwjvMfpakb4fjzh7pdVsMaspoCP16Ji11cBdaJf/m0J/6L+f2ATWhRbp3ium9y9YbHLiSkeO7/Y9XmEsNuK8IM8r+iOKLDnE/aIIPyQdyME0kTCHuCBlB9IRR4ETiL8Ipc8YFqZ1wy/fQ/rR9dTvZZmhP8+is+kN4+yXQgYMDlqOZxSmRqi6wXA1u7+FbCCEJ4HEPaavzWztpT/fqbaXvHvbfHrFVHW828L3FLs+7OY8L4057fK+1kqWd+86Gsq4rti138q5fP6AGbWxMweNbMFZrac8LO3Fen7PeGP090VrC/rKeCTMR+YUyKwG7j7YRl6/pbFrrci7KH+AHxL+CUGIOoztiS0dGBDwB8QXZ9IxQL+ScIvy2x3Lxmo5b3mik5rmuq1LASal+ijtirridx9kbv/xd2bEfZi7yytr11eDdE2fmFDEE0ktIdqR3udEwnHJrYAPkrj+UtaCLQo9nnLsh4Yqeh7Oh84rcT3qK67v1vKY8v7WSpZX6voaypTV3n+Hj3nHu6+GdCP8IcpXSOBfwEvRscI8oYCPhmTgeVmdpGZ1TWzmmbWzsw6Zej5+5nZrma2KXA1MC5qEz0O/N7MuptZLWAI8DNQ9As8ETgIqOvu3xD+DT6U0J/8sLyNRi2hg4HShruV95q/A1pHLY10pHot7xGC9mwzKzCzo4F9ynoiM/ujmRUF5xJCWKxLo4axwLlmtp2Z1Se0NB6L2hsQ3s+/EnrCEP4jGkRo3aXz/CU9DpxsZrtE39vLy3n8d4TjA+m6G7jYzHYDMLPNzeyPKWpJ9bMEcJaZtbBwkP8SoOjg9XdAI4sOSGdAA8J/S0vNrDlwQSWe46+EFuPzZlY3Q3UlTgGfgOiXuzfh3/c5hL3re4FUP/AX2q/Hwf+Q4rFjCAePFhEOUp0dbXcWYe/mtmibvQnDO9dE939B+EV5K/p8OeFA6DvpBpK7T43aExV9zU9EH/9rJcbUl7GdMl9L9HqOJrSLlhD6xeNTPF0n4AMzWwE8C5zj7ukM8byP8F5Pil7TakKAF5lICJ+igH+bcHB0EpXg7i8BtxL65l8S/pBBCNbS3AIcE41CuTWN538KuAF4NGp1zAB6lfHYlD9LkUeAVwg/Q7MJBzNx988JfxxnR+2girZuSroK2AtYBrxA6u91qaJ23kDCfzHPWLGRS7nMft2mFJFcEQ1LnAFsUuy/hqxg4QS3P7v7a0nXUp1pD14kh5jZUWZW28y2IOxtP5dt4S7ZQwEvkltOI4xd/4pwnOCMZMuRbKYWjYhIntIevIhInipIuoDittpqK2/dunXSZYiI5Ixp06b94O6NS7svqwK+devWTJ06NekyRERyhpmVeZa2WjQiInlKAS8ikqcU8CIieUoBLyKSpxTwIiJ5SgEvIpKnFPAiInkq5wN+3Tq4/nqYkmpBNhGRaijnA37FCrjzTujXD1atSroaEZHskfMBv/nmMHo0fPEFXHRR0tWIiGSPnA94gIMPhsGD4fbb4ZVXkq5GRCQ75EXAA1x3Hey6K5x8MixenHQ1IiLJy5uAr1sXHnoICgvhzDOTrkZEJHl5E/AAHTrAlVfCY4/B2LFJVyMikqy8CniACy+ELl3CXvz8+UlXIyKSnLwL+IICePBBWLs29OPXr0+6IhGRZORdwANsvz0MHw6vvx5G1oiIVEd5GfAAf/4zHH54GBs/c2bS1YiIVL28DXgzGDkS6tcPZ7muWZN0RSIiVStvAx5gm23gnntg+nS45pqkqxERqVp5HfAARx0FJ50UToR6//2kqxERqTp5H/AAt9wCLVtC//6wcmXS1YiIVI1qEfCbbQYPPABffQXnn590NSIiVaNaBDzAgQfCkCFw993w4otJVyMiEr9YA97MzjWzT81shpmNNbM6cW6vPNdeC7vvDqecAj/8kGQlIiLxiy3gzaw5cDbQ0d3bATWB4+LaXjo22SRMSLZkCZx2GrgnWY2ISLzibtEUAHXNrADYFPg25u2Va489wpDJ8eNhzJikqxERiU9sAe/uC4CbgK+BhcAyd8+K5TiGDIEDDoBBg2DevKSrERGJR5wtmi2APsB2QDOgnpn1K+VxA81sqplNLSwsjKucX6lZM4yqWb8+jJHXhGQiko/ibNH0AOa4e6G7rwXGA11KPsjd73H3ju7esXHjxjGW82vbbQe33goTJsCIEVW2WRGRKhNnwH8NdDazTc3MgO5AVk37ddJJ0KcPXHwxzJiRdDUiIpkVZw/+A2AcMB34JNrWPXFtrzLMwlw1DRuGCcl+/jnpikREMifWUTTufoW77+zu7dy9v7tnXYQ2aQL33gsffxyW+xMRyRfV5kzWVHr3DvPH33ADvP120tWIiGSGAj4ybBi0bg0DBsCPPyZdjYjIxlPARxo0CCc+zZsH556bdDUiIhtPAV/MfvuFJf5GjYJnn026GhGRjaOAL+HKK6F9+9CT//77pKsREak8BXwJtWuHVs3y5TBwoCYkE5HcpYAvRbt2YYm/Z56B++9PuhoRkcpRwJdh8GA46CA45xyYMyfpakREKk4BX4YaNWD06PBxwABYty7pikREKkYBn0KrVnD77eHkp5tuSroaEZGKUcCXo18/+MMfYOjQMJ2BiEiuUMCXwyws1N2oUQj71auTrkhEJD0K+DRstRXcd1+YUnjo0KSrERFJjwI+Tb16wemnw803w8SJSVcjIlI+BXwF3HQTbL99GFWzbFnS1YiIpKaAr4B69cJZrt98E8bHi4hkMwV8BXXuDJdeGhbtHj8+6WpERMqmgK+EoUNh773DXDWLFiVdjYhI6RTwlVCrVmjVrFwJp56qCclEJDsp4Ctpl13CEn8vvggjRyZdjYjIbyngN8Jf/wo9eoQVoL78MulqRER+TQG/EWrUCNMJ164dhk7+8kvSFYmIbKCA30gtWsCdd8J774WWjYhItlDAZ0DfvnDssWG5v+nTk65GRCRQwGfInXdCkyZhQrKffkq6GhERBXzGbLll6MfPnAmXXJJ0NSIiCviM6tkzjKwZMQJefz3pakSkulPAZ9gNN0DbtnDSSbB0adLViEh1poDPsE03DWe5LlwY9uZFRJKigI9Bp05w+eXw8MPw+ONJVyMi1ZUCPiaXXAL77BMWCVmwIOlqRKQ6UsDHpKAgtGpWr9aEZCKSDAV8jHbaKawC9fLLcNddSVcjItWNAj5mZ5wB//M/cP75MGtW0tWISHWigI+ZGdx3H9StC/37w9q1SVckItWFAr4KNGsGd98NU6bAddclXY2IVBcK+Cryxz/CCSfANdfA5MlJVyMi1YECvgrdfjs0bRqCXmu5ikjcYgt4M2trZh8Vuyw3s8FxbS8XNGwIY8fCt9/CAQfA3LlJVyQi+Sy2gHf3We7e3t3bA3sDq4Cn4tperth/f3jtNfjhh3D9s8+SrkhE8lVVtWi6A1+5+7wq2l5W23dfmDgxLPHXtStMnZp0RSKSj6oq4I8DxpZ2h5kNNLOpZja1sLCwispJ3h57wNtvQ4MGcPDBMGFC0hWJSL6JPeDNrDZwBPBEafe7+z3u3tHdOzZu3DjucrLKDjuEkG/ZEg49FJ57LumKRCSfVMUefC9gurt/VwXbyjnNm8OkSWGP/qij4KGHkq5IRPJFVQR8X8poz0jQqFFYAapr13C26x13JF2RiOSDWAPezDYFDgHGx7mdfNCgAbz4IhxxRFgo5NprNQOliGycWAPe3Ve5eyN3XxbndvJFnTrw5JNhL37o0DBBmUJeRCqrIOkC5NcKCmD06HBS1LBhYV3Xf/4z3C4iUhGKjSxUowbccgtssQVcfXUI+UcegU02SboyEcklmosmS5nBVVfB8OEwfjz07g0rVyZdlYjkEgV8lhs8GO6/P4yyOeQQWLIk6YpEJFco4HPASSfBE0/AtGlw4IGaiVJE0pN2wJtZTTNrZmatii5xFia/dvTR8MILMHt2mKRszpykKxKRbJdWwJvZIOA74FXghejyfIx1SSl69AgzUS5erJkoRaR86e7BnwO0dffd3H336LJHnIVJ6Tp3DjNRrl8fznydMiXpikQkW6Ub8PMBnayUJXbfPUxSttlmYSbKN99MuiIRyUbpjoOfDUwwsxeAn4tudPdhsVQl5dp++xDyhxwCvXrB44+HaQ5ERIqkuwf/NaH/XhtoUOwiCWrWLMxEueee4SDsmDFJVyQi2SStPXh3vwrAzBqET31FrFVJ2ho1CgdejzwSBgwIZ70OGpR0VSKSDdIdRdPOzD4EZgCfmtk0M9st3tIkXQ0ahCGURx4JZ58N11yjScpEJP0WzT3Aee6+rbtvCwwBRsZXllRUnTrhZKgBA+Dyy2HIkDDSRkSqr3QPstZz9/8fq+HuE8ysXkw1SSUVFIRpDRo2DHPYLFkCI0dqJkqR6irtUTRmNhQoOozXD9C5lFmoRg0YMQK23BKuvBKWL9dMlCLVVbotmlOAxoSVmZ6Krp8cV1GycczgiitC0I8fD4cfDit0WFyk2kl3FM0S4OyYa5EMO+ec0K459dQwXv6FF8KevYhUDykD3sxGuPtgM3sO+M24DHfXqTVZ7sQTYfPN4dhjw0yUr7wCTZsmXZWIVIXy9uCLeu43xV2IxOfII8OC3n36hEnKXnsNttsu6apEJG4pe/DuPi262t7dJxa/AO3jL08ypXt3eOONcCLUfvvBp58mXZGIxC3dg6wnlnLbSRmsQ6rAPvuEmSghzEQ5eXKy9YhIvFIGvJn1jfrvbczs2WKXN4H/Vk2Jkknt2oVJyho23LBXLyL5qbwe/LvAQmAr4OZit/8I/DuuoiRebdqEkO/ZEw47DB57LPTnRSS/lNeDnwe8Baws0YOf7u6/VE2JEoemTUO7pn17+MMf4MEHk65IRDKt3B68u68DVpnZ5lVQj1ShLbcMI2q6dQvDKW+7LemKRCST0p2qYDXwiZm9CqwsutHddfJTjqtfP5wA1bdvmIlyyRIYOjScDSsiuS3dgC9aaFvy0CabhBWh/vKXMMXB4sUwbFiY10ZEcle6UxU8YGa1gZ2im2a5+9r4ypKqVlAAo0aF0TUjRsCyZZqJUiTXpfXra2bdgAeAuYABLc3sRHefFF9pUtVq1Ah77ltuGeaUX7oUxo4Nc82LSO5Jd//sZqCnu88CMLOdgLHA3nEVJskwCz34hg1DT/7ww+Gpp8KqUSKSW9LtstYqCncAd/8CqBVPSZINBg0KQycnTAgnRz3/fNIViUhFpRvwU81slJl1iy4jgWnlfpXktP79w1j5Bg2gd2845hhYsCDpqkQkXekG/BnAp4Q54c8BPgNOi6soyR777QfTp8N114XhlLvsEsbLr1uXdGUiUp50A/50dx/m7ke7+1HuPpwQ+lIN1K4NF18cZqDs0iX05jt3DsEvItlLs0lK2tq0gZdegkcfhfnzoVMnOPdc+PHHpCsTkdKkO5vkdiVmk5yAZpOslszC6lCffw4DB8Itt8Cuu8LTTyddmYiUpNkkpVIaNoS77gpz2Jx2Ghx1VJiR8rbboGXLpKsTEUhjNkl3nwD0AN6KVnJaCLQgnPCUkpk1NLNxZva5mc00s30zUbRkj86dYepUuPFGePXVcBB2+HD4RXONiiQu3R78JKCOmTUHXgdOBkan8XW3AP9y952BPYGZlSlSslutWnDBBeEgbLducN55YfWoKVOSrkykeks34M3dVwFHA7e5+1HArim/wGwzoCswCsDd17j70o0pVrJb69bw3HMwbhwsWgS/+104YWr58qQrE6me0g74qL1yAhtmlSyvf98GKATuN7MPzexeM6tXyhMPNLOpZja1sLAw7cIlO5mFBURmzoSzzoI77ghtm3HjwD3p6kSql3QDfjBwMfCUu39qZm2AN8v5mgJgL+Aud+9AmEf+f0s+yN3vcfeO7t6xcePGFShdstnmm4cDru+/D02awB//GM6GnTs36cpEqo+0Aj5apu8Id78h+nx2Got9fAN84+4fRJ+PIwS+VCNFvfhhw8K8NrvtBv/4B6zVZNMisStvHPyI6ONzJcbBP2tmz6b6WndfBMw3s7bRTd0JUxxINVNQEE6I+uwzOOQQuPBC6Ngx7N2LSHzK66OPiT7eVMnnHwQ8HC0WMpsw+kaqqVatwglRTz8dDr526QKnnx7muWnYMOnqRPKPeZpHvsysMYC7x3YktGPHjj516tS4nl6yyI8/hkVFbr019OhHjIA//UlrwYpUlJlNc/eOpd1XXovGzOxKM/sB+Bz4wswKzezyOAqV6qNBg3BC1JQp0Lw5HHcc9OoFs2cnXZlI/ijvIOtgYD+gk7s3cvctgN8B+5nZubFXJ3lvr73ggw/Cnvy774aDsNdfr4OwIplQXsAPAPq6+5yiG9x9NtAvuk9ko9WsGXryM2fC738fpibu0AHeeSfpykRyW3kBX8vdfyh5Y9SH15J9klHNm4cTop59NvTo998/zFi5eHHSlYnkpvICfk0l7xOptN69w7w2558P990HO+8MDz+sM2FFKqq8gN/TzJaXcvkR2L0qCpTqqX79cELUtGlhoZF+/aBnT/jPf5KuTCR3lDddcE1336yUSwN3V4tGYrfnnqEXf8cdMHky7L47XHst/Pxz0pWJZL9056IRSUzNmnDmmWEVqT59YOhQaN8eJk1KujKR7KaAl5zRtCk89hi8+CKsXg0HHginnAL/1eKRIqVSwEvO6dUrHIS96CIYMyYchH3gAR2EFSlJAS85adNNwwlR06fDTjvBSSfBwQfDrFlJVyaSPRTwktN23x3eegv++U/46KPw+QUXwLJlSVcmkjwFvOS8GjXCCVGffw79+8PNN8OOO8LIkbBuXdLViSRHAS95Y+utYdSoMIFZ27Yh9Dt2hIkTk65MJBkKeMk7e+8dhlA++miY5qBbt7BkoJYLlOpGAS95yQyOPTa0ba6+Ogyt3HlnuPRSWLEi6epEqoYCXvJa3brhxKhZs+CYY8LqUTvtFIZVrl+fdHUi8VLAS7XQogU89BC89x60bBmGVXbuHD4XyVcKeKlWikL9wQdhwYKwLuwJJ8A33yRdmUjmKeCl2qlRIwynnDUr9OSffDK0ba6+GlatSro6kcxRwEu1Vb9+mJny88/h8MPhiivCgdhHH9W0B5IfFPBS7bVuDY8/HsbLb7UV9O0LBxwAU6cmXZnIxlHAi0S6dg0nSd17b1hYZJ99wmyVixYlXZlI5SjgRYqpWRNOPTUE/Pnnh5E3O+4YJjZbvTrp6kQqRgEvUorNNoMbb4TPPoPu3eHii2HXXeGpp9Sfl9yhgBdJYYcd4Omn4dVXwxTFRx8dAv/f/066MpHyKeBF0tCjR5iO+I47Qrh36ACnnw6FhUlXJlI2BbxImgoKwtqw//kPDBoUDsbuuCMMGwZr1iRdnchvKeBFKmiLLWDECPjkE9h3XxgyJCw08sIL6s9LdlHAi1TSLrvASy+FYDcLJ0v16gUzZyZdmUiggBfZSIcdFvryw4fD+++Hvfmzzw5z0YskSQEvkgG1a8PgwaE//5e/hIOxO+4YPv7yS9LVSXWlgBfJoMaN4a674MMPoX17+Otfw8dXX026MqmOFPAiMdhjD3jttXBi1E8/Qc+e0KdP2MMXqSoKeJGYmMGRR4azYa+/Ht54A3bbDS64AJYtS7o6qQ4U8CIx22QTuOiisPfevz/cfHPoz48cqf68xEsBL1JFttkGRo0KM1a2bQsDB4alBIcMCWfJagy9ZJoCXqSK7b03TJoEzz4blgy87bYw9cEee8A//hGWEhTJhFgD3szmmtknZvaRmWn5BJGIGfTuDePHw8KFcOedYYWpCy8Mi4L37AljxsCKFUlXKrmsKvbgD3L39u7esQq2JZJzGjWCM84Ii4F/8QVcdhl8+SUMGBDaOgMGhBE569YlXankGrVoRLLIjjuGxb+/+greeguOPz60cg45BFq1Cnv4M2YkXaXkCvMYj+yY2RxgCeDAP939nlIeMxAYCNCqVau9582bF1s9Irlo9Wp47rnQsnnppTDypn37sGfft2/Yy5fqy8ymldUhiTvgm7n7t2bWBHgVGOTuk8p6fMeOHX2qVjoWKVNhITz6aAj7KVPCEoM9e4bhl336hEVJpHpJFfCxtmjc/dvo4/fAU8A+cW5PJN81bhzmop88OcxaedFFoWVz/PFhT/6UU+DNN2H9+qQrlWwQW8CbWT0za1B0HegJqHsokiE77wx/+xvMnRtC/ZhjYNw4OPhg2G47uOQSTV1c3cW5B7818LaZfQxMBl5w93/FuD2RaqlGDejWDe67DxYtgrFjw5QIN94YFgrv1CmMtdfygtVPrD34ilIPXiRzisJ+zJgwu2VBQViQpH//MAa/Tp2kK5RMSKwHLyLJ2WYbOPdcmD49LC943nnh+p/+FO4bODAMxcyifTzJMAW8SDXQrh3ccAPMmxfmpu/TBx55BLp2he23h8sv11TG+UgBL1KN1KwJPXrAAw/Ad9+F9s2OO4aDtTvtFBYRv/NO+O9/k65UMkEBL1JN1asH/frByy/D/PlhorOVK+Gss6BpUzjqqLBgyc8/J12pVJYCXkRo1gzOPz8sHv7RR2Gs/fvvw9FHh7A/88wwV4769blFAS8iv7LnnmFRkvnzw9QIvXrB6NFhauM2bcL4+k8+SbpKSYcCXkRKVVAAhx4KDz8c+vUPPhhOrrrxxjB3/e67w3XXwZw5SVcqZVHAi0i5GjQI4+dfeinMX3/HHdCwIVx6adir33dfuPXWMPZesocCXkQqpHHj0JN/660wTcINN8BPP8E550Dz5mFq4/vug6VLk65UFPAiUmnbbhvmqP/oI/j009CfnzMHTj0Vtt46jMR54onwB0CqngJeRDJi113hmmvCCVMffBD28j/4IJw526RJmL/+pZdg7dqkK60+FPAiklFmsM8+MHx4GInz+utw3HFh0ZLDDgtDMotaPJrWOF4KeBGJTc2aYfrikSPDAdhnnoHu3cOwy65doXXrDS0ejbHPPAW8iFSJTTaBI44IK1J9/z089FAYbjl8OHTosKHF8+WXSVeaPxTwIlLl6teHE06A558Pe/Z33x0Oyl5xRZgbp1OnEPzffpt0pblNAS8iiWrUCE47DSZMgK+/hptuCr35886DFi02tHgWL0660tyjgBeRrNGiBQwZAtOmweefh2mMFywIc9dvs01o8YwdGyZFk/Ip4EUkK7VtC1deGYJ+2jQ4++ywYMnxx4dhl8cfH0bmrFmTdKXZSwEvIlnNDPbaK7Ruvv46tHL69w/THE1bPXgAAAcgSURBVB9xxIbVqSZM0LDLkhTwIpIzatSAAw8MB2UXLgwHaXv1CqtTHXQQtGwZWjxTp2rYJWjRbRHJAytXhrB/5JENZ8tuvXU4QFt0adMm6SrjkWrRbQW8iOSVJUvg6afhtdfgjTc2zHDZuvWvA79p00TLzBgFvIhUS+4wc2YI+jfegDff3DDL5S67hKDv3j20fbbcMtlaK0sBLyICrFsXpkUoCvxJk2DVqnAgt0OHDYG///7hZKxcoIAXESnFmjUwefKGwH/vvXBbQQH87nch7A8+GDp3DlMtZCMFvIhIGlatgnfeCWH/+uth/P369VC3btirL+rf77VX+COQDRTwIiKVsHRpaOO8/noI/Rkzwu2bbx769kUtnd12C22eJCjgRUQy4LvvwoHaopbOV1+F25s0CePwiwK/TZuqC3wFvIhIDObN2xD2r78eTr4CaNVqQ9gfdFBYqzYuCngRkZi5w6xZvx6SWTQDZtu2Gw7YdusWZtDMFAW8iEgVW78ePv7410MyV6wIrZs999wQ+AccAA0aVH47CngRkYStXQtTpmxo57z77oYhmV26hNtr1qz486YK+CwZ6CMikt9q1QpB3qULXHYZ/PRTCPk33ghLGFYm3MujgBcRSUDduqFN0717fNvQdMEiInlKAS8ikqcU8CIieUoBLyKSp2IPeDOraWYfmtnzcW9LREQ2qIo9+HOAmVWwHRERKSbWgDezFsDvgXvj3I6IiPxW3HvwI4ALgfVlPcDMBprZVDObWlhYGHM5IiLVR2wnOpnZ4cD37j7NzLqV9Th3vwe4J/qaQjObV8lNbgX8UMmvrWq5VCvkVr25VCvkVr25VCvkVr0bU+u2Zd0R21w0ZvZ3oD/wC1AH2AwY7+79Ytre1LLmY8g2uVQr5Fa9uVQr5Fa9uVQr5Fa9cdUaW4vG3S929xbu3ho4DngjrnAXEZHf0jh4EZE8VSWTjbn7BGBCzJu5J+bnz6RcqhVyq95cqhVyq95cqhVyq95Yas2q+eBFRCRz1KIREclTCngRkTyV0wFvZi3N7E0zm2lmn5rZOUnXlIqZ1TGzyWb2cVTvVUnXVJ5cmkvIzOaa2Sdm9pGZZfXaj2bW0MzGmdnn0c/vvknXVBYzaxu9p0WX5WY2OOm6ymJm50a/XzPMbKyZ1Um6plTM7Jyo1k8z/b7mdA/ezJoCTd19upk1AKYBR7r7ZwmXViozM6Ceu68ws1rA28A57v5+wqWVyczOAzoCm7n74UnXk4qZzQU6unvWn9xiZg8Ab7n7vWZWG9jU3ZcmXVd5zKwmsAD4nbtX9qTE2JhZc8Lv1a7u/pOZPQ686O6jk62sdGbWDngU2AdYA/wLOMPd/5OJ58/pPXh3X+ju06PrPxImNWuebFVl82BF9Gmt6JK1f2E1l1A8zGwzoCswCsDd1+RCuEe6A19lY7gXUwDUNbMCYFPg24TrSWUX4H13X+XuvwATgaMy9eQ5HfDFmVlroAPwQbKVpBa1PD4CvgdedfdsrrfcuYSyjAOvmNk0MxuYdDEptAEKgfuj9te9ZlYv6aLSdBwwNukiyuLuC4CbgK+BhcAyd38l2apSmgF0NbNGZrYpcBjQMlNPnhcBb2b1gSeBwe6+POl6UnH3de7eHmgB7BP9i5Z1is8llHQtFbCfu+8F9ALOMrOuSRdUhgJgL+Aud+8ArAT+N9mSyhe1ko4Anki6lrKY2RZAH2A7oBlQz8yy9gx6d58J3AC8SmjPfEyY3iUjcj7go172k8DD7j4+6XrSFf1LPgE4NOFSyrIfcETU134UONjMHkq2pNTc/dvo4/fAU4S+Zjb6Bvim2H9v4wiBn+16AdPd/bukC0mhBzDH3QvdfS0wHuiScE0pufsod9/L3bsCi4GM9N8hxwM+Omg5Cpjp7sOSrqc8ZtbYzBpG1+sSfhg/T7aq0uXaXEJmVi860E7U7uhJ+Pc367j7ImC+mbWNbuoOZOXAgBL6ksXtmcjXQGcz2zTKh+5k+YJDZtYk+tgKOJoMvsdVMlVBjPYjzFj5SdTXBrjE3V9MsKZUmgIPRCMRagCPu3vWDz/MEVsDT4XfaQqAR9z9X8mWlNIg4OGo7TEbODnhelKK+sOHAKclXUsq7v6BmY0DphNaHR+S/VMWPGlmjYC1wFnuviRTT5zTwyRFRKRsOd2iERGRsingRUTylAJeRCRPKeBFRPKUAl5EJE8p4EVSMLPWZpaV4+lFyqOAFxHJUwp4kTSZWZtocrBOSdcikg4FvEgaomkFngROdvcpSdcjko5cn6pApCo0Bp4B/uDunyZdjEi6tAcvUr5lwHzC3EciOUN78CLlWwMcCbxsZivc/ZGkCxJJhwJeJA3uvjJaBOVVM1vp7s8kXZNIeTSbpIhInlIPXkQkTyngRUTylAJeRCRPKeBFRPKUAl5EJE8p4EVE8pQCXkQkT/0fzECvj05LvHQAAAAASUVORK5CYII=","text/plain":["<Figure size 432x288 with 1 Axes>"]},"metadata":{"needs_background":"light"},"output_type":"display_data"}],"source":["X_line = [K[0], K[-1]]\n","Y_line = [distortions[0], distortions[-1]]\n","\n","# Plot the elbow\n","plt.plot(K, distortions, 'b-')\n","#plt.plot(X_line, Y_line, 'r')\n","plt.xlabel('k')\n","plt.ylabel('Distortion')\n","plt.title('The Elbow Method showing the optimal k')\n","plt.show()"]},{"cell_type":"code","execution_count":50,"metadata":{"executionInfo":{"elapsed":350,"status":"ok","timestamp":1664477517717,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"P-nt53CUlAFf"},"outputs":[{"name":"stdout","output_type":"stream","text":["6       7\n","16      7\n","27      5\n","31      5\n","35      3\n","       ..\n","4301    3\n","4317    3\n","4343    1\n","4351    7\n","4362    4\n","Name: y, Length: 398, dtype: int32\n"]},{"name":"stderr","output_type":"stream","text":["/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n","/opt/anaconda3/lib/python3.7/site-packages/sklearn/cluster/k_means_.py:445: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\n","Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","  max_iter=max_iter, verbose=verbose)\n"]}],"source":["k=8\n","kmeans=KMeans(n_clusters=k,random_state=42)\n","\n","y_pred=kmeans.fit_predict(X_embedded)\n","data['y']=y_pred\n","print(data['y'])"]},{"cell_type":"code","execution_count":36,"metadata":{"executionInfo":{"elapsed":1,"status":"ok","timestamp":1664477519794,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"ATtnnlyHr-nu"},"outputs":[],"source":["vectorizers = []\n","    \n","for ii in range(0,8 ):\n","    # Creating a vectorizer\n","    vectorizers.append(CountVectorizer(min_df=0, max_df=0.9))"]},{"cell_type":"code","execution_count":37,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":351,"status":"ok","timestamp":1664477522681,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"SOXjcjiVsC_7","outputId":"c9aea2be-b2c4-4d5e-eacb-4294916196e1"},"outputs":[],"source":["vectorized_data = []\n","\n","for current_cluster, cvec in enumerate(vectorizers):\n","    try:\n","        vectorized_data.append(cvec.fit_transform(data.loc[data['y'] == current_cluster, 'cleaned']))\n","    except Exception as e:\n","        print(\"Not enough instances in cluster: \" + str(current_cluster))\n","        vectorized_data.append(None)"]},{"cell_type":"code","execution_count":38,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":475,"status":"ok","timestamp":1664477526857,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"wTQhiDtnsWnq","outputId":"97b7c9d9-64d8-43e2-d0c3-f01c4e661a74"},"outputs":[{"data":{"text/plain":["LatentDirichletAllocation(batch_size=128, doc_topic_prior=None,\n","                          evaluate_every=-1, learning_decay=0.7,\n","                          learning_method='online', learning_offset=10.0,\n","                          max_doc_update_iter=100, max_iter=10,\n","                          mean_change_tol=0.001, n_components=8, n_jobs=None,\n","                          perp_tol=0.1, random_state=42, topic_word_prior=None,\n","                          total_samples=1000000.0, verbose=False)"]},"execution_count":38,"metadata":{},"output_type":"execute_result"}],"source":["#Topic modeling will be performed using the LDA. \n","#This is a generateve statistical model that \n","#allows sets of words to be explained by a shared topic. \n","NUM_TOPICS_PER_CLUSTER = 8\n","\n","\n","lda_models = []\n","\n","for ii in range(0, 8):\n","    # Latent Dirichlet Allocation Model\n","    lda = LatentDirichletAllocation(n_components=NUM_TOPICS_PER_CLUSTER, max_iter=10, learning_method='online',verbose=False, random_state=42)\n","    lda_models.append(lda)\n","    \n","lda_models[0]"]},{"cell_type":"code","execution_count":39,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":1048,"status":"ok","timestamp":1664477530476,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"GKgxTddAskEn","outputId":"7506409d-b81b-4eb0-be9a-ef9654631eb1"},"outputs":[{"name":"stdout","output_type":"stream","text":["Current Cluster: 0\n","Current Cluster: 1\n","Current Cluster: 2\n","Current Cluster: 3\n","Current Cluster: 4\n","Current Cluster: 5\n","Current Cluster: 6\n","Current Cluster: 7\n"]}],"source":["#For each clyuster, we have created a corresponding LDA model. We will\n","#now fir_transform all the LDA models on their respective cluter vectors. \n","\n","\n","clusters_lda_data = []\n","\n","for current_cluster, lda in enumerate(lda_models):\n","    print(\"Current Cluster: \" + str(current_cluster))\n","    \n","    if vectorized_data[current_cluster] != None:\n","        clusters_lda_data.append((lda.fit_transform(vectorized_data[current_cluster])))"]},{"cell_type":"code","execution_count":40,"metadata":{"executionInfo":{"elapsed":213,"status":"ok","timestamp":1664477532777,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"5r84nO0DwTrq"},"outputs":[],"source":["\n","def select_topics(model, vectorizer, top_n=3):\n","    curr_words = []\n","    keywords = []\n","    \n","    for idx, topic in enumerate(model.components_):\n","        words = [(vectorizer.get_feature_names()[i], topic[i]) for i in topic.argsort()[:-top_n - 1:-1]]\n","        for word in words:\n","            if word[0] not in curr_words:\n","                keywords.append(word)\n","                curr_words.append(word[0])\n","                \n","    keywords.sort(key = lambda x: x[1])  \n","    keywords.reverse()\n","    return_values = []\n","    for i in keywords:\n","        return_values.append(i[0])\n","    return return_values"]},{"cell_type":"code","execution_count":41,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":229,"status":"ok","timestamp":1664477535955,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"DDMmhxTGwV-Q","outputId":"cc45a4b4-e031-4048-dafa-dac3ad86a9ed"},"outputs":[{"name":"stdout","output_type":"stream","text":["Current Cluster: 0\n","Current Cluster: 1\n","Current Cluster: 2\n","Current Cluster: 3\n","Current Cluster: 4\n","Current Cluster: 5\n","Current Cluster: 6\n","Current Cluster: 7\n"]}],"source":["all_keywords = []\n","for current_vectorizer, lda in enumerate(lda_models):\n","    print(\"Current Cluster: \" + str(current_vectorizer))\n","\n","    if vectorized_data[current_vectorizer] != None:\n","        all_keywords.append(select_topics(lda, vectorizers[current_vectorizer]))"]},{"cell_type":"code","execution_count":42,"metadata":{"executionInfo":{"elapsed":497,"status":"ok","timestamp":1664477538681,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"o_fsmGbXwYAe"},"outputs":[{"data":{"text/html":["\n","    <div class=\"bk-root\">\n","        <a href=\"https://bokeh.pydata.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n","        <span id=\"1176\">Loading BokehJS ...</span>\n","    </div>"]},"metadata":{},"output_type":"display_data"},{"data":{"application/javascript":"\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  var JS_MIME_TYPE = 'application/javascript';\n  var HTML_MIME_TYPE = 'text/html';\n  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n  var CLASS_NAME = 'output_bokeh rendered_html';\n\n  /**\n   * Render data to the DOM node\n   */\n  function render(props, node) {\n    var script = document.createElement(\"script\");\n    node.appendChild(script);\n  }\n\n  /**\n   * Handle when an output is cleared or removed\n   */\n  function handleClearOutput(event, handle) {\n    var cell = handle.cell;\n\n    var id = cell.output_area._bokeh_element_id;\n    var server_id = cell.output_area._bokeh_server_id;\n    // Clean up Bokeh references\n    if (id != null && id in Bokeh.index) {\n      Bokeh.index[id].model.document.clear();\n      delete Bokeh.index[id];\n    }\n\n    if (server_id !== undefined) {\n      // Clean up Bokeh references\n      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n      cell.notebook.kernel.execute(cmd, {\n        iopub: {\n          output: function(msg) {\n            var id = msg.content.text.trim();\n            if (id in Bokeh.index) {\n              Bokeh.index[id].model.document.clear();\n              delete Bokeh.index[id];\n            }\n          }\n        }\n      });\n      // Destroy server and session\n      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n      cell.notebook.kernel.execute(cmd);\n    }\n  }\n\n  /**\n   * Handle when a new output is added\n   */\n  function handleAddOutput(event, handle) {\n    var output_area = handle.output_area;\n    var output = handle.output;\n\n    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n      return\n    }\n\n    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n\n    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n      // store reference to embed id on output_area\n      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n    }\n    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n      var bk_div = document.createElement(\"div\");\n      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n      var script_attrs = bk_div.children[0].attributes;\n      for (var i = 0; i < script_attrs.length; i++) {\n        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n      }\n      // store reference to server id on output_area\n      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n    }\n  }\n\n  function register_renderer(events, OutputArea) {\n\n    function append_mime(data, metadata, element) {\n      // create a DOM node to render to\n      var toinsert = this.create_output_subarea(\n        metadata,\n        CLASS_NAME,\n        EXEC_MIME_TYPE\n      );\n      this.keyboard_manager.register_events(toinsert);\n      // Render to node\n      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n      render(props, toinsert[toinsert.length - 1]);\n      element.append(toinsert);\n      return toinsert\n    }\n\n    /* Handle when an output is cleared or removed */\n    events.on('clear_output.CodeCell', handleClearOutput);\n    events.on('delete.Cell', handleClearOutput);\n\n    /* Handle when a new output is added */\n    events.on('output_added.OutputArea', handleAddOutput);\n\n    /**\n     * Register the mime type and append_mime function with output_area\n     */\n    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n      /* Is output safe? */\n      safe: true,\n      /* Index of renderer in `output_area.display_order` */\n      index: 0\n    });\n  }\n\n  // register the mime type if in Jupyter Notebook environment and previously unregistered\n  if (root.Jupyter !== undefined) {\n    var events = require('base/js/events');\n    var OutputArea = require('notebook/js/outputarea').OutputArea;\n\n    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n      register_renderer(events, OutputArea);\n    }\n  }\n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"1176\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"1176\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '1176' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.3.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.3.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.3.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.3.4.min.js\"];\n  var css_urls = [];\n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    \n    function(Bokeh) {\n      \n    },\n    function(Bokeh) {} // ensure no trailing comma for IE\n  ];\n\n  function run_inline_js() {\n    \n    if ((root.Bokeh !== undefined) || (force === true)) {\n      for (var i = 0; i < inline_js.length; i++) {\n        inline_js[i].call(root, root.Bokeh);\n      }if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"1176\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));","application/vnd.bokehjs_load.v0+json":""},"metadata":{},"output_type":"display_data"}],"source":["# show on notebook\n","output_notebook()\n","# target labels\n","y_labels = y_pred"]},{"cell_type":"code","execution_count":43,"metadata":{"executionInfo":{"elapsed":480,"status":"ok","timestamp":1664477543955,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"9sbGhh1dm8X0"},"outputs":[],"source":["source = ColumnDataSource(data=dict(\n","    x= X_embedded[:,0], \n","    y= X_embedded[:,1],\n","    x_backup = X_embedded[:,0],\n","    y_backup = X_embedded[:,1],\n","    desc= y_labels, \n","    titles= data['titles'],\n","    journal = data['periodicals'],\n","    links = data['urls'],\n","    labels = [\"C-\" + str(x) for x in y_labels]\n","    ))\n","\n","\n","# hover over information\n","hover = HoverTool(tooltips=[\n","    (\"Title\", \"@titles{safe}\"),\n","    (\"Journal\", \"@journal\"),\n","    (\"Link\", \"@links\")\n","],\n","point_policy=\"follow_mouse\")\n","\n","# map colors\n","mapper = linear_cmap(field_name='desc', \n","                     palette=Category20[20],\n","                     low=min(y_labels) ,high=max(y_labels))"]},{"cell_type":"code","execution_count":44,"metadata":{"executionInfo":{"elapsed":1,"status":"ok","timestamp":1664477546475,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"rOLwDSZJnGZ_"},"outputs":[],"source":["# prepare the figure\n","plot = figure(plot_width=1200, plot_height=850, \n","           tools=[hover, 'pan', 'wheel_zoom', 'box_zoom', 'reset', 'save', 'tap'], \n","           title=\"Clustering of the COVID-19 Literature\", \n","           toolbar_location=\"above\")"]},{"cell_type":"code","execution_count":45,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":2,"status":"ok","timestamp":1664477549203,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"HMgx-Cr-p9Gr","outputId":"aade5823-c193-488d-bffc-9664152cef99"},"outputs":[],"source":["\n","# plot settings\n","plot.scatter('x', 'y', size=5, \n","          source=source,\n","          fill_color=mapper,\n","          line_alpha=0.3,\n","          legend = 'labels')\n","plot.legend.background_fill_alpha = 0.6"]},{"cell_type":"code","execution_count":53,"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":887},"executionInfo":{"elapsed":323,"status":"ok","timestamp":1664477553812,"user":{"displayName":"Vanessa Ilana Klotzman","userId":"15432622816603113078"},"user_tz":420},"id":"N5drYzLMqEfT","outputId":"44ed11b4-0598-41a8-b90e-3accd9d00815"},"outputs":[{"data":{"text/html":["\n","\n","\n","\n","\n","\n","  <div class=\"bk-root\" id=\"dbcfc130-9bb4-45f6-b42f-0183c785354e\" data-root-id=\"1865\"></div>\n"]},"metadata":{},"output_type":"display_data"},{"data":{"application/javascript":"(function(root) {\n  function embed_document(root) {\n    \n  var docs_json = {\"842baa40-3731-4afa-bd8a-a9dac2d13a25\":{\"roots\":{\"references\":[{\"attributes\":{\"children\":[{\"id\":\"1864\",\"type\":\"Row\"}],\"sizing_mode\":\"scale_both\"},\"id\":\"1865\",\"type\":\"Column\"},{\"attributes\":{\"ticker\":{\"id\":\"1192\",\"type\":\"BasicTicker\"}},\"id\":\"1195\",\"type\":\"Grid\"},{\"attributes\":{\"formatter\":{\"id\":\"1222\",\"type\":\"BasicTickFormatter\"},\"ticker\":{\"id\":\"1192\",\"type\":\"BasicTicker\"}},\"id\":\"1191\",\"type\":\"LinearAxis\"},{\"attributes\":{\"source\":{\"id\":\"1177\",\"type\":\"ColumnDataSource\"}},\"id\":\"1219\",\"type\":\"CDSView\"},{\"attributes\":{},\"id\":\"1244\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"text\":\"Clustering of the COVID-19 Literature\"},\"id\":\"1181\",\"type\":\"Title\"},{\"attributes\":{},\"id\":\"1205\",\"type\":\"SaveTool\"},{\"attributes\":{},\"id\":\"1189\",\"type\":\"LinearScale\"},{\"attributes\":{},\"id\":\"1224\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"formatter\":{\"id\":\"1224\",\"type\":\"BasicTickFormatter\"},\"ticker\":{\"id\":\"1197\",\"type\":\"BasicTicker\"}},\"id\":\"1196\",\"type\":\"LinearAxis\"},{\"attributes\":{},\"id\":\"1222\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"callback\":null},\"id\":\"1206\",\"type\":\"TapTool\"},{\"attributes\":{\"callback\":null},\"id\":\"1183\",\"type\":\"DataRange1d\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"1178\",\"type\":\"HoverTool\"},{\"id\":\"1201\",\"type\":\"PanTool\"},{\"id\":\"1202\",\"type\":\"WheelZoomTool\"},{\"id\":\"1203\",\"type\":\"BoxZoomTool\"},{\"id\":\"1204\",\"type\":\"ResetTool\"},{\"id\":\"1205\",\"type\":\"SaveTool\"},{\"id\":\"1206\",\"type\":\"TapTool\"}]},\"id\":\"1207\",\"type\":\"Toolbar\"},{\"attributes\":{},\"id\":\"1202\",\"type\":\"WheelZoomTool\"},{\"attributes\":{},\"id\":\"1204\",\"type\":\"ResetTool\"},{\"attributes\":{\"bottom_units\":\"screen\",\"fill_alpha\":{\"value\":0.5},\"fill_color\":{\"value\":\"lightgrey\"},\"left_units\":\"screen\",\"level\":\"overlay\",\"line_alpha\":{\"value\":1.0},\"line_color\":{\"value\":\"black\"},\"line_dash\":[4,4],\"line_width\":{\"value\":2},\"render_mode\":\"css\",\"right_units\":\"screen\",\"top_units\":\"screen\"},\"id\":\"1225\",\"type\":\"BoxAnnotation\"},{\"attributes\":{\"background_fill_alpha\":{\"value\":0.6},\"items\":[{\"id\":\"1227\",\"type\":\"LegendItem\"}]},\"id\":\"1226\",\"type\":\"Legend\"},{\"attributes\":{\"callback\":null},\"id\":\"1185\",\"type\":\"DataRange1d\"},{\"attributes\":{\"callback\":null,\"point_policy\":\"follow_mouse\",\"tooltips\":[[\"Title\",\"@titles{safe}\"],[\"Journal\",\"@journal\"],[\"Link\",\"@links\"]]},\"id\":\"1178\",\"type\":\"HoverTool\"},{\"attributes\":{\"label\":{\"field\":\"labels\"},\"renderers\":[{\"id\":\"1218\",\"type\":\"GlyphRenderer\"}]},\"id\":\"1227\",\"type\":\"LegendItem\"},{\"attributes\":{\"dimension\":1,\"ticker\":{\"id\":\"1197\",\"type\":\"BasicTicker\"}},\"id\":\"1200\",\"type\":\"Grid\"},{\"attributes\":{\"overlay\":{\"id\":\"1225\",\"type\":\"BoxAnnotation\"}},\"id\":\"1203\",\"type\":\"BoxZoomTool\"},{\"attributes\":{\"high\":7,\"low\":0,\"palette\":[\"#1f77b4\",\"#aec7e8\",\"#ff7f0e\",\"#ffbb78\",\"#2ca02c\",\"#98df8a\",\"#d62728\",\"#ff9896\",\"#9467bd\",\"#c5b0d5\",\"#8c564b\",\"#c49c94\",\"#e377c2\",\"#f7b6d2\",\"#7f7f7f\",\"#c7c7c7\",\"#bcbd22\",\"#dbdb8d\",\"#17becf\",\"#9edae5\"]},\"id\":\"1179\",\"type\":\"LinearColorMapper\"},{\"attributes\":{},\"id\":\"1192\",\"type\":\"BasicTicker\"},{\"attributes\":{\"callback\":null,\"data\":{\"desc\":{\"__ndarray__\":\"BAAAAAIAAAAGAAAAAwAAAAcAAAAGAAAAAwAAAAIAAAAFAAAAAgAAAAEAAAADAAAAAAAAAAAAAAAHAAAAAQAAAAEAAAAHAAAAAQAAAAAAAAAGAAAAAQAAAAAAAAAAAAAABgAAAAUAAAADAAAAAwAAAAIAAAABAAAABQAAAAQAAAAEAAAABwAAAAIAAAAGAAAAAwAAAAcAAAAGAAAAAwAAAAIAAAAFAAAAAgAAAAEAAAADAAAAAAAAAAAAAAAHAAAAAQAAAAEAAAAHAAAAAQAAAAAAAAAGAAAABQAAAAEAAAAAAAAAAAAAAAQAAAAGAAAABQAAAAYAAAAAAAAABAAAAAIAAAACAAAAAwAAAAMAAAACAAAABgAAAAEAAAAFAAAABAAAAAYAAAAEAAAABwAAAAIAAAAEAAAABgAAAAMAAAAHAAAABgAAAAMAAAAAAAAAAwAAAAIAAAADAAAABQAAAAIAAAABAAAAAwAAAAAAAAAAAAAABwAAAAEAAAABAAAABwAAAAEAAAAAAAAABgAAAAUAAAABAAAAAAAAAAAAAAAEAAAABgAAAAUAAAADAAAABgAAAAMAAAAAAAAABAAAAAIAAAAHAAAAAgAAAAMAAAADAAAAAgAAAAIAAAAGAAAAAQAAAAcAAAAFAAAABAAAAAIAAAACAAAAAgAAAAQAAAAHAAAAAwAAAAEAAAAEAAAABAAAAAQAAAABAAAABwAAAAcAAAAEAAAABAAAAAMAAAAHAAAAAAAAAAAAAAADAAAAAAAAAAUAAAAAAAAAAgAAAAcAAAAGAAAAAAAAAAMAAAACAAAABQAAAAYAAAACAAAAAAAAAAEAAAAAAAAAAgAAAAMAAAAAAAAABgAAAAAAAAAGAAAABgAAAAMAAAAGAAAAAQAAAAMAAAADAAAABgAAAAEAAAACAAAAAwAAAAIAAAABAAAABQAAAAcAAAACAAAAAgAAAAUAAAAGAAAABAAAAAcAAAACAAAABAAAAAYAAAADAAAABwAAAAYAAAADAAAAAAAAAAMAAAACAAAAAwAAAAUAAAACAAAAAAAAAAQAAAABAAAAAwAAAAAAAAAAAAAABwAAAAEAAAAEAAAAAQAAAAcAAAABAAAAAAAAAAYAAAAFAAAAAQAAAAAAAAAAAAAABAAAAAYAAAAFAAAAAwAAAAAAAAAGAAAAAgAAAAMAAAAFAAAAAAAAAAIAAAAEAAAAAgAAAAcAAAACAAAABAAAAAMAAAADAAAABQAAAAIAAAACAAAABgAAAAEAAAAHAAAABQAAAAQAAAAGAAAABAAAAAcAAAACAAAABAAAAAYAAAADAAAAAwAAAAYAAAAHAAAABgAAAAAAAAADAAAAAwAAAAUAAAAAAAAAAwAAAAIAAAADAAAABQAAAAIAAAAAAAAAAwAAAAQAAAABAAAAAwAAAAAAAAAAAAAABwAAAAEAAAAEAAAABQAAAAEAAAADAAAABwAAAAEAAAAAAAAABgAAAAUAAAAFAAAAAQAAAAAAAAAAAAAABwAAAAQAAAAGAAAABQAAAAMAAAAAAAAABgAAAAIAAAADAAAABQAAAAAAAAAAAAAAAgAAAAAAAAAEAAAAAgAAAAcAAAACAAAAAwAAAAQAAAADAAAAAwAAAAUAAAADAAAAAgAAAAIAAAAGAAAAAQAAAAcAAAAFAAAABAAAAAMAAAAEAAAABgAAAAQAAAAHAAAAAwAAAAIAAAAEAAAABgAAAAMAAAADAAAABgAAAAcAAAAGAAAAAAAAAAMAAAADAAAABQAAAAAAAAADAAAAAgAAAAMAAAAFAAAAAgAAAAAAAAADAAAABAAAAAEAAAADAAAAAAAAAAAAAAAHAAAAAQAAAAQAAAADAAAABQAAAAEAAAADAAAABwAAAAEAAAAAAAAABgAAAAUAAAAFAAAAAQAAAAMAAAAAAAAAAAAAAAcAAAAEAAAABgAAAAUAAAADAAAABAAAAAAAAAAGAAAAAgAAAAMAAAAFAAAAAAAAAAAAAAACAAAAAAAAAAQAAAACAAAABwAAAAIAAAADAAAABAAAAAMAAAADAAAABQAAAAMAAAACAAAAAgAAAAYAAAABAAAABwAAAAUAAAAEAAAAAwAAAAQAAAA=\",\"dtype\":\"int32\",\"shape\":[398]},\"journal\":[\"medRxiv\",\"Journal of travel medicine\",\"Emerging microbes & infections\",\"Microbes and infection\",\"Journal of medical virology\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Acta M\\u00e9dica Portuguesa\",\"Journal of Medical Virology\",\"Viruses\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Journal of medical virology\",\"Military Medical Research\",\"Epidemiology and Health\",\"Epidemiology and Health\",\"Journal of Korean medical science\",\"Journal of biological regulators and homeostatic agents\",\"Journal of medical virology\",\"International Health\",\"Frontiers in microbiology\",\"arXiv preprint arXiv:2002.00426\",\"Zhonghua yi xue za zhi\",\"Veterinary Quarterly\",\"Journal of clinical medicine\",\"Journal of Clinical Medicine\",\"Infection, Genetics and Evolution\",\"Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"Zhonghua nei ke za zhi\",\"Journal of medical virology\",\"Microbes and infection\",\"bioRxiv\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"arXiv preprint arXiv:2001.10530\",\"medRxiv\",\"Pathogens and Global Health\",\"Journal of travel medicine\",\"Emerging microbes & infections\",\"Microbes and infection\",\"Journal of medical virology\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Acta M\\u00e9dica Portuguesa\",\"Journal of Medical Virology\",\"Viruses\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Journal of medical virology\",\"Military Medical Research\",\"Epidemiology and Health\",\"Epidemiology and Health\",\"Journal of Korean medical science\",\"Journal of biological regulators and homeostatic agents\",\"Journal of medical virology\",\"International Health\",\"Frontiers in microbiology\",\"arXiv preprint arXiv:2002.00426\",\"Zhonghua yi xue za zhi\",\"Journal of Travel Medicine\",\"Veterinary Quarterly\",\"Journal of clinical medicine\",\"Journal of Clinical Medicine\",\"European Radiology\",\"Infection, Genetics and Evolution\",\"Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"Viruses\",\"Journal of Clinical Medicine\",\"Communicable diseases intelligence (2018)\",\"Journal of Clinical Medicine\",\"Clinical Infectious Diseases\",\"Zhonghua nei ke za zhi\",\"Journal of medical virology\",\"Microbes and infection\",\"Journal of Medical Virology\",\"bioRxiv\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"arXiv preprint arXiv:2001.10530\",\"Wellcome Open Research\",\"medRxiv\",\"Pathogens and Global Health\",\"Journal of travel medicine\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 571\",\"Emerging microbes & infections\",\"Microbes and infection\",\"Journal of medical virology\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Journal of Clinical Medicine\",\"medRxiv\",\"Acta M\\u00e9dica Portuguesa\",\"Journal of Medical Virology\",\"Biomolecules\",\"Viruses\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Journal of medical virology\",\"Military Medical Research\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 523\",\"Epidemiology and Health\",\"Journal of Korean medical science\",\"Journal of biological regulators and homeostatic agents\",\"Journal of medical virology\",\"International Health\",\"Frontiers in microbiology\",\"arXiv preprint arXiv:2002.00426\",\"Zhonghua yi xue za zhi\",\"Journal of Travel Medicine\",\"Veterinary Quarterly\",\"Journal of clinical medicine\",\"Journal of Clinical Medicine\",\"European Radiology\",\"Infection, Genetics and Evolution\",\"Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"Journal of Infection in Developing Countries\",\"Viruses\",\"Clinical and Translational Medicine\",\"Journal of Clinical Medicine\",\"Communicable diseases intelligence (2018)\",\"Journal of Clinical Medicine\",\"Emerging Microbes & Infections\",\"Clinical Infectious Diseases\",\"Zhonghua nei ke za zhi\",\"Journal of medical virology\",\"Microbes and infection\",\"The Journal of Infectious Diseases\",\"Journal of Medical Virology\",\"bioRxiv\",\"Emerging Microbes & Infections\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"arXiv preprint arXiv:2001.10530\",\"British Journal of Nursing\",\"British Journal of Nursing\",\"Pathogens\",\"medRxiv\",\"International Health\",\"Microbes and infection\",\"Journal of medical virology\",\"arXiv preprint arXiv:2001.10530\",\"Communicable diseases intelligence (2018)\",\"Preprints (www.preprints.org)\",\"bioRxiv\",\"Journal of medical virology\",\"Emerging Microbes & Infections\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 571\",\"European Radiology\",\"Journal of medical virology\",\"Emerging Microbes & Infections\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 523\",\"medRxiv\",\"Journal of Clinical Medicine\",\"Journal of Clinical Medicine\",\"Viruses\",\"Journal of clinical medicine\",\"Journal of Medical Virology\",\"Journal of Korean medical science\",\"Emerging microbes & infections\",\"Journal of Clinical Medicine\",\"Biomolecules\",\"The Journal of Infectious Diseases\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Zhonghua yi xue za zhi\",\"Journal of Clinical Medicine\",\"Epidemiology and Health\",\"Frontiers in microbiology\",\"Journal of Clinical Medicine\",\"Journal of travel medicine\",\"Acta M\\u00e9dica Portuguesa\",\"arXiv preprint arXiv:2002.00426\",\"Viruses\",\"Epidemiology and Infection\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Infection, Genetics and Evolution\",\"Clinical and Translational Medicine\",\"Wellcome Open Research\",\"Veterinary Quarterly\",\"Zhonghua nei ke za zhi\",\"Military Medical Research\",\"Journal of Medical Virology\",\"Journal of biological regulators and homeostatic agents\",\"Infectious Disease Modelling\",\"Journal of Infection in Developing Countries\",\"Microbes and infection\",\"Journal of medical virology\",\"Journal of Travel Medicine\",\"Pathogens and Global Health\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Clinical Infectious Diseases\",\"Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"Wellcome Open Research\",\"medRxiv\",\"Pathogens and Global Health\",\"Journal of travel medicine\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 571\",\"Emerging microbes & infections\",\"Microbes and infection\",\"Journal of medical virology\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Journal of Clinical Medicine\",\"medRxiv\",\"Acta M\\u00e9dica Portuguesa\",\"Journal of Medical Virology\",\"Biomolecules\",\"Viruses\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Epidemiology and Infection\",\"Preprints (www.preprints.org)\",\"Journal of medical virology\",\"Military Medical Research\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 523\",\"Epidemiology and Health\",\"Journal of Korean medical science\",\"Journal of biological regulators and homeostatic agents\",\"medRxiv\",\"Journal of medical virology\",\"International Health\",\"Frontiers in microbiology\",\"arXiv preprint arXiv:2002.00426\",\"Zhonghua yi xue za zhi\",\"Journal of Travel Medicine\",\"Veterinary Quarterly\",\"Journal of clinical medicine\",\"Journal of Clinical Medicine\",\"European Radiology\",\"Infection, Genetics and Evolution\",\"Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"Journal of Infection in Developing Countries\",\"Journal of Clinical Medicine\",\"Viruses\",\"Pathogens\",\"Clinical and Translational Medicine\",\"Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"Journal of Clinical Medicine\",\"Infectious Disease Modelling\",\"Communicable diseases intelligence (2018)\",\"Journal of Clinical Medicine\",\"Emerging Microbes & Infections\",\"Clinical Infectious Diseases\",\"Cell Discovery\",\"Zhonghua nei ke za zhi\",\"Journal of medical virology\",\"Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"Microbes and infection\",\"The Journal of Infectious Diseases\",\"Journal of Medical Virology\",\"bioRxiv\",\"Emerging Microbes & Infections\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"arXiv preprint arXiv:2001.10530\",\"Wellcome Open Research\",\"medRxiv\",\"Pathogens and Global Health\",\"Journal of travel medicine\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 571\",\"Emerging microbes & infections\",\"Microbes and infection\",\"The Journal of infection\",\"F1000Research\",\"Journal of medical virology\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Epidemiology and Health\",\"Journal of Clinical Medicine\",\"F1000Research\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"medRxiv\",\"Acta M\\u00e9dica Portuguesa\",\"Journal of Medical Virology\",\"Biomolecules\",\"Viruses\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Epidemiology and Infection\",\"F1000Research\",\"Preprints (www.preprints.org)\",\"Journal of medical virology\",\"Military Medical Research\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 523\",\"Epidemiology and Health\",\"Journal of Korean medical science\",\"Journal of biological regulators and homeostatic agents\",\"medRxiv\",\"Journal of travel medicine\",\"Journal of medical virology\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"International Health\",\"Frontiers in microbiology\",\"arXiv preprint arXiv:2002.00426\",\"Zhonghua yi xue za zhi\",\"The Journal of infection\",\"Journal of Travel Medicine\",\"Veterinary Quarterly\",\"Journal of clinical medicine\",\"Journal of Clinical Medicine\",\"F1000Research\",\"European Radiology\",\"Infection, Genetics and Evolution\",\"Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"Journal of Infection in Developing Countries\",\"Journal of Clinical Medicine\",\"Viruses\",\"Pathogens\",\"Clinical and Translational Medicine\",\"Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"Zhonghua yu fang yi xue za zhi [Chinese journal of preventive medicine]\",\"Journal of Clinical Medicine\",\"Infectious Disease Modelling\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Communicable diseases intelligence (2018)\",\"Journal of Clinical Medicine\",\"Emerging Microbes & Infections\",\"Clinical Infectious Diseases\",\"F1000Research\",\"Cell Discovery\",\"Zhonghua nei ke za zhi\",\"Journal of medical virology\",\"Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"The Journal of infection\",\"Microbes and infection\",\"The Journal of Infectious Diseases\",\"Journal of Medical Virology\",\"bioRxiv\",\"Emerging Microbes & Infections\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"arXiv preprint arXiv:2001.10530\",\"Preprints (www.preprints.org)\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Wellcome Open Research\",\"medRxiv\",\"Pathogens and Global Health\",\"Data in Brief\",\"Journal of travel medicine\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 571\",\"Emerging microbes & infections\",\"Microbes and infection\",\"The Journal of infection\",\"F1000Research\",\"Journal of medical virology\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Epidemiology and Health\",\"Journal of Clinical Medicine\",\"F1000Research\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"medRxiv\",\"Acta M\\u00e9dica Portuguesa\",\"Journal of Medical Virology\",\"Biomolecules\",\"Viruses\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Epidemiology and Infection\",\"F1000Research\",\"Preprints (www.preprints.org)\",\"Journal of medical virology\",\"Military Medical Research\",\"Journal of Clinical Medicine 2020, Vol. 9, Page 523\",\"Epidemiology and Health\",\"Journal of Korean medical science\",\"Journal of biological regulators and homeostatic agents\",\"medRxiv\",\"Journal of Travel Medicine\",\"Journal of travel medicine\",\"Journal of medical virology\",\"Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"International Health\",\"Frontiers in microbiology\",\"arXiv preprint arXiv:2002.00426\",\"Zhonghua yi xue za zhi\",\"The Journal of infection\",\"Journal of Travel Medicine\",\"Veterinary Quarterly\",\"Infectious Disease Modelling\",\"Journal of clinical medicine\",\"Journal of Clinical Medicine\",\"F1000Research\",\"European Radiology\",\"Infection, Genetics and Evolution\",\"Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"Journal of Infection in Developing Countries\",\"Journal of Travel Medicine\",\"Journal of Clinical Medicine\",\"Viruses\",\"Pathogens\",\"Clinical and Translational Medicine\",\"Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"Zhonghua yu fang yi xue za zhi [Chinese journal of preventive medicine]\",\"Journal of Clinical Medicine\",\"Infectious Disease Modelling\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Communicable diseases intelligence (2018)\",\"Journal of Clinical Medicine\",\"Emerging Microbes & Infections\",\"Clinical Infectious Diseases\",\"F1000Research\",\"Cell Discovery\",\"Zhonghua nei ke za zhi\",\"Journal of medical virology\",\"Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"The Journal of infection\",\"Microbes and infection\",\"The Journal of Infectious Diseases\",\"Journal of Medical Virology\",\"bioRxiv\",\"Emerging Microbes & Infections\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"arXiv preprint arXiv:2001.10530\",\"Preprints (www.preprints.org)\",\"International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\"],\"labels\":[\"C-4\",\"C-2\",\"C-6\",\"C-3\",\"C-7\",\"C-6\",\"C-3\",\"C-2\",\"C-5\",\"C-2\",\"C-1\",\"C-3\",\"C-0\",\"C-0\",\"C-7\",\"C-1\",\"C-1\",\"C-7\",\"C-1\",\"C-0\",\"C-6\",\"C-1\",\"C-0\",\"C-0\",\"C-6\",\"C-5\",\"C-3\",\"C-3\",\"C-2\",\"C-1\",\"C-5\",\"C-4\",\"C-4\",\"C-7\",\"C-2\",\"C-6\",\"C-3\",\"C-7\",\"C-6\",\"C-3\",\"C-2\",\"C-5\",\"C-2\",\"C-1\",\"C-3\",\"C-0\",\"C-0\",\"C-7\",\"C-1\",\"C-1\",\"C-7\",\"C-1\",\"C-0\",\"C-6\",\"C-5\",\"C-1\",\"C-0\",\"C-0\",\"C-4\",\"C-6\",\"C-5\",\"C-6\",\"C-0\",\"C-4\",\"C-2\",\"C-2\",\"C-3\",\"C-3\",\"C-2\",\"C-6\",\"C-1\",\"C-5\",\"C-4\",\"C-6\",\"C-4\",\"C-7\",\"C-2\",\"C-4\",\"C-6\",\"C-3\",\"C-7\",\"C-6\",\"C-3\",\"C-0\",\"C-3\",\"C-2\",\"C-3\",\"C-5\",\"C-2\",\"C-1\",\"C-3\",\"C-0\",\"C-0\",\"C-7\",\"C-1\",\"C-1\",\"C-7\",\"C-1\",\"C-0\",\"C-6\",\"C-5\",\"C-1\",\"C-0\",\"C-0\",\"C-4\",\"C-6\",\"C-5\",\"C-3\",\"C-6\",\"C-3\",\"C-0\",\"C-4\",\"C-2\",\"C-7\",\"C-2\",\"C-3\",\"C-3\",\"C-2\",\"C-2\",\"C-6\",\"C-1\",\"C-7\",\"C-5\",\"C-4\",\"C-2\",\"C-2\",\"C-2\",\"C-4\",\"C-7\",\"C-3\",\"C-1\",\"C-4\",\"C-4\",\"C-4\",\"C-1\",\"C-7\",\"C-7\",\"C-4\",\"C-4\",\"C-3\",\"C-7\",\"C-0\",\"C-0\",\"C-3\",\"C-0\",\"C-5\",\"C-0\",\"C-2\",\"C-7\",\"C-6\",\"C-0\",\"C-3\",\"C-2\",\"C-5\",\"C-6\",\"C-2\",\"C-0\",\"C-1\",\"C-0\",\"C-2\",\"C-3\",\"C-0\",\"C-6\",\"C-0\",\"C-6\",\"C-6\",\"C-3\",\"C-6\",\"C-1\",\"C-3\",\"C-3\",\"C-6\",\"C-1\",\"C-2\",\"C-3\",\"C-2\",\"C-1\",\"C-5\",\"C-7\",\"C-2\",\"C-2\",\"C-5\",\"C-6\",\"C-4\",\"C-7\",\"C-2\",\"C-4\",\"C-6\",\"C-3\",\"C-7\",\"C-6\",\"C-3\",\"C-0\",\"C-3\",\"C-2\",\"C-3\",\"C-5\",\"C-2\",\"C-0\",\"C-4\",\"C-1\",\"C-3\",\"C-0\",\"C-0\",\"C-7\",\"C-1\",\"C-4\",\"C-1\",\"C-7\",\"C-1\",\"C-0\",\"C-6\",\"C-5\",\"C-1\",\"C-0\",\"C-0\",\"C-4\",\"C-6\",\"C-5\",\"C-3\",\"C-0\",\"C-6\",\"C-2\",\"C-3\",\"C-5\",\"C-0\",\"C-2\",\"C-4\",\"C-2\",\"C-7\",\"C-2\",\"C-4\",\"C-3\",\"C-3\",\"C-5\",\"C-2\",\"C-2\",\"C-6\",\"C-1\",\"C-7\",\"C-5\",\"C-4\",\"C-6\",\"C-4\",\"C-7\",\"C-2\",\"C-4\",\"C-6\",\"C-3\",\"C-3\",\"C-6\",\"C-7\",\"C-6\",\"C-0\",\"C-3\",\"C-3\",\"C-5\",\"C-0\",\"C-3\",\"C-2\",\"C-3\",\"C-5\",\"C-2\",\"C-0\",\"C-3\",\"C-4\",\"C-1\",\"C-3\",\"C-0\",\"C-0\",\"C-7\",\"C-1\",\"C-4\",\"C-5\",\"C-1\",\"C-3\",\"C-7\",\"C-1\",\"C-0\",\"C-6\",\"C-5\",\"C-5\",\"C-1\",\"C-0\",\"C-0\",\"C-7\",\"C-4\",\"C-6\",\"C-5\",\"C-3\",\"C-0\",\"C-6\",\"C-2\",\"C-3\",\"C-5\",\"C-0\",\"C-0\",\"C-2\",\"C-0\",\"C-4\",\"C-2\",\"C-7\",\"C-2\",\"C-3\",\"C-4\",\"C-3\",\"C-3\",\"C-5\",\"C-3\",\"C-2\",\"C-2\",\"C-6\",\"C-1\",\"C-7\",\"C-5\",\"C-4\",\"C-3\",\"C-4\",\"C-6\",\"C-4\",\"C-7\",\"C-3\",\"C-2\",\"C-4\",\"C-6\",\"C-3\",\"C-3\",\"C-6\",\"C-7\",\"C-6\",\"C-0\",\"C-3\",\"C-3\",\"C-5\",\"C-0\",\"C-3\",\"C-2\",\"C-3\",\"C-5\",\"C-2\",\"C-0\",\"C-3\",\"C-4\",\"C-1\",\"C-3\",\"C-0\",\"C-0\",\"C-7\",\"C-1\",\"C-4\",\"C-3\",\"C-5\",\"C-1\",\"C-3\",\"C-7\",\"C-1\",\"C-0\",\"C-6\",\"C-5\",\"C-5\",\"C-1\",\"C-3\",\"C-0\",\"C-0\",\"C-7\",\"C-4\",\"C-6\",\"C-5\",\"C-3\",\"C-4\",\"C-0\",\"C-6\",\"C-2\",\"C-3\",\"C-5\",\"C-0\",\"C-0\",\"C-2\",\"C-0\",\"C-4\",\"C-2\",\"C-7\",\"C-2\",\"C-3\",\"C-4\",\"C-3\",\"C-3\",\"C-5\",\"C-3\",\"C-2\",\"C-2\",\"C-6\",\"C-1\",\"C-7\",\"C-5\",\"C-4\",\"C-3\",\"C-4\"],\"links\":[\"internal-pdf://https://www.medrxiv.org/content/medrxiv/early/2020/01/30/2020.01.27.20018986.full.pdfhttp://orcid.org/0000-0003-4169-5665\",\"https://www.ncbi.nlm.nih.gov/pubmed/31943059\",\"https://www.ncbi.nlm.nih.gov/pubmed/31987001\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32032682\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967327\",\"http://www.ncbi.nlm.nih.gov/pubmed/32035430\",\"https://actamedicaportuguesa.com/revista/index.php/amp/article/view/13547\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25699\",\"https://www.mdpi.com/1999-4915/12/2/135\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033514\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967321\",\"https://www.ncbi.nlm.nih.gov/pubmed/32029004\",\"http://www.ncbi.nlm.nih.gov/pubmed/32035431http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020007\",\"internal-pdf://Ki - Unknown - Epidemiologic characteristics of early cases with 2019-nCoV disease in Republ.pdfhttp://www.ncbi.nlm.nih.gov/pubmed/32035431http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020007\",\"https://www.ncbi.nlm.nih.gov/pubmed/32030925\",\"https://www.ncbi.nlm.nih.gov/pubmed/32013309\",\"https://www.ncbi.nlm.nih.gov/pubmed/31981224\",\"internal-pdf://https://academic.oup.com/inthealth/advance-article-abstract/doi/10.1093/inthealth/ihaa001/5732900\",\"https://www.ncbi.nlm.nih.gov/pubmed/32038520https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6992538/\",\"https://ui.adsabs.harvard.edu/abs/2020arXiv200200426Lhttp://arxiv.org/abs/2002.00426\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033513\",\"internal-pdf://https://www.tandfonline.com/action/journalInformation?journalCode=tveq20https://www.tandfonline.com/doi/full/10.1080/01652176.2020.1727993\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32033064\",\"https://www.mdpi.com/2077-0383/9/2/330\",\"https://www.biorxiv.org/content/biorxiv/early/2020/01/27/2020.01.26.920249.full.pdf\",\"https://www.ncbi.nlm.nih.gov/pubmed/32007627\",\"http://www.ncbi.nlm.nih.gov/pubmed/32023681\",\"https://www.ncbi.nlm.nih.gov/pubmed/32031264\",\"https://www.ncbi.nlm.nih.gov/pubmed/32017984\",\"internal-pdf://Zhang et al. - Unknown - Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replication Structure-based design.pdfhttp://dx.doi.org/10.1101/2020.02.10.936898http://biorxiv.org/content/early/2020/02/10/2020.02.10.936898.abstract\",\"internal-pdf://Zhao et al. - 2020 - Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 202.pdfhttps://www.ncbi.nlm.nih.gov/pubmed/32007643https://www.clinicalkey.com/#!/content/playContent/1-s2.0-S1201971220300539?returnurl=https:%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1201971220300539%3Fshowall%3Dtrue&referrer=https:%2F%2Fwww.ncbi.nlm.\",\"internal-pdf://\",\"internal-pdf://https://www.medrxiv.org/content/medrxiv/early/2020/01/30/2020.01.27.20018986.full.pdfhttp://orcid.org/0000-0003-4169-5665\",\"https://www.tandfonline.com/doi/full/10.1080/20477724.2020.1725339\",\"https://www.ncbi.nlm.nih.gov/pubmed/31943059\",\"https://www.ncbi.nlm.nih.gov/pubmed/31987001\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32032682\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967327\",\"http://www.ncbi.nlm.nih.gov/pubmed/32035430\",\"https://actamedicaportuguesa.com/revista/index.php/amp/article/view/13547\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25699\",\"https://www.mdpi.com/1999-4915/12/2/135\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033514\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967321\",\"https://www.ncbi.nlm.nih.gov/pubmed/32029004\",\"http://www.ncbi.nlm.nih.gov/pubmed/32035431http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020007\",\"internal-pdf://Ki - Unknown - Epidemiologic characteristics of early cases with 2019-nCoV disease in Republ.pdfhttp://www.ncbi.nlm.nih.gov/pubmed/32035431http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020007\",\"https://www.ncbi.nlm.nih.gov/pubmed/32030925\",\"https://www.ncbi.nlm.nih.gov/pubmed/32013309\",\"https://www.ncbi.nlm.nih.gov/pubmed/31981224\",\"internal-pdf://https://academic.oup.com/inthealth/advance-article-abstract/doi/10.1093/inthealth/ihaa001/5732900\",\"https://www.ncbi.nlm.nih.gov/pubmed/32038520https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6992538/\",\"https://ui.adsabs.harvard.edu/abs/2020arXiv200200426Lhttp://arxiv.org/abs/2002.00426\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033513\",\"internal-pdf://https://academic.oup.com/jtm/advance-article/doi/10.1093/jtm/taaa021/5735319\",\"internal-pdf://https://www.tandfonline.com/action/journalInformation?journalCode=tveq20https://www.tandfonline.com/doi/full/10.1080/01652176.2020.1727993\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32033064\",\"https://www.mdpi.com/2077-0383/9/2/330\",\"internal-pdf://http://link.springer.com/10.1007/s00330-020-06731-x\",\"https://www.biorxiv.org/content/biorxiv/early/2020/01/27/2020.01.26.920249.full.pdf\",\"https://www.ncbi.nlm.nih.gov/pubmed/32007627\",\"internal-pdf://https://www.mdpi.com/1999-4915/12/2/194\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/462\",\"https://www.ncbi.nlm.nih.gov/pubmed/32050080\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/498\",\"internal-pdf://https://academic.oup.com/cid/advance-article/doi/10.1093/cid/ciaa149/5734265\",\"http://www.ncbi.nlm.nih.gov/pubmed/32023681\",\"https://www.ncbi.nlm.nih.gov/pubmed/32031264\",\"https://www.ncbi.nlm.nih.gov/pubmed/32017984\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25707\",\"internal-pdf://Zhang et al. - Unknown - Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replication Structure-based design.pdfhttp://dx.doi.org/10.1101/2020.02.10.936898http://biorxiv.org/content/early/2020/02/10/2020.02.10.936898.abstract\",\"internal-pdf://Zhao et al. - 2020 - Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 202.pdfhttps://www.ncbi.nlm.nih.gov/pubmed/32007643https://www.clinicalkey.com/#!/content/playContent/1-s2.0-S1201971220300539?returnurl=https:%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1201971220300539%3Fshowall%3Dtrue&referrer=https:%2F%2Fwww.ncbi.nlm.\",\"internal-pdf://\",\"internal-pdf://https://wellcomeopenresearch.org/articles/5-17/v1\",\"internal-pdf://https://www.medrxiv.org/content/medrxiv/early/2020/01/30/2020.01.27.20018986.full.pdfhttp://orcid.org/0000-0003-4169-5665\",\"https://www.tandfonline.com/doi/full/10.1080/20477724.2020.1725339\",\"https://www.ncbi.nlm.nih.gov/pubmed/31943059\",\"internal-pdf://\",\"https://www.ncbi.nlm.nih.gov/pubmed/31987001\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32032682\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967327\",\"http://www.ncbi.nlm.nih.gov/pubmed/32035430\",\"http://www.ncbi.nlm.nih.gov/pubmed/32093211https://www.mdpi.com/2077-0383/9/2/575\",\"internal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan.pdfinternal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan(2).pdfhttps://www.medrxiv.org/content/10.1101/2020.01.28.20019299v3.full.pdf\",\"https://actamedicaportuguesa.com/revista/index.php/amp/article/view/13547\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25699\",\"internal-pdf://https://www.mdpi.com/2218-273X/10/2/331\",\"https://www.mdpi.com/1999-4915/12/2/135\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033514\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967321\",\"https://www.ncbi.nlm.nih.gov/pubmed/32029004\",\"internal-pdf://\",\"internal-pdf://Ki - Unknown - Epidemiologic characteristics of early cases with 2019-nCoV disease in Republ.pdfhttp://www.ncbi.nlm.nih.gov/pubmed/32035431http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020007\",\"https://www.ncbi.nlm.nih.gov/pubmed/32030925\",\"https://www.ncbi.nlm.nih.gov/pubmed/32013309\",\"https://www.ncbi.nlm.nih.gov/pubmed/31981224\",\"internal-pdf://https://academic.oup.com/inthealth/advance-article-abstract/doi/10.1093/inthealth/ihaa001/5732900\",\"https://www.ncbi.nlm.nih.gov/pubmed/32038520https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6992538/\",\"https://ui.adsabs.harvard.edu/abs/2020arXiv200200426Lhttp://arxiv.org/abs/2002.00426\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033513\",\"internal-pdf://https://academic.oup.com/jtm/advance-article/doi/10.1093/jtm/taaa021/5735319\",\"internal-pdf://https://www.tandfonline.com/action/journalInformation?journalCode=tveq20https://www.tandfonline.com/doi/full/10.1080/01652176.2020.1727993\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32033064\",\"https://www.mdpi.com/2077-0383/9/2/330\",\"internal-pdf://http://link.springer.com/10.1007/s00330-020-06731-x\",\"https://www.biorxiv.org/content/biorxiv/early/2020/01/27/2020.01.26.920249.full.pdf\",\"https://www.ncbi.nlm.nih.gov/pubmed/32007627\",\"internal-pdf://https://jidc.org/index.php/journal/article/view/12425\",\"internal-pdf://https://www.mdpi.com/1999-4915/12/2/194\",\"internal-pdf://https://clintransmed.springeropen.com/articles/10.1186/s40169-020-00271-z\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/462\",\"https://www.ncbi.nlm.nih.gov/pubmed/32050080\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/498\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729069\",\"internal-pdf://https://academic.oup.com/cid/advance-article/doi/10.1093/cid/ciaa149/5734265\",\"http://www.ncbi.nlm.nih.gov/pubmed/32023681\",\"https://www.ncbi.nlm.nih.gov/pubmed/32031264\",\"https://www.ncbi.nlm.nih.gov/pubmed/32017984\",\"internal-pdf://https://academic.oup.com/jid/advance-article/doi/10.1093/infdis/jiaa077/5739751\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25707\",\"internal-pdf://Zhang et al. - Unknown - Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replication Structure-based design.pdfhttp://dx.doi.org/10.1101/2020.02.10.936898http://biorxiv.org/content/early/2020/02/10/2020.02.10.936898.abstract\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729071\",\"internal-pdf://Zhao et al. - 2020 - Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 202.pdfhttps://www.ncbi.nlm.nih.gov/pubmed/32007643https://www.clinicalkey.com/#!/content/playContent/1-s2.0-S1201971220300539?returnurl=https:%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1201971220300539%3Fshowall%3Dtrue&referrer=https:%2F%2Fwww.ncbi.nlm.\",\"internal-pdf://\",\"internal-pdf://https://www.health.qld.gov.au/__data/assets/pdf_file/0025/444490/icare-pivc-guideline.pdf\",\"internal-pdf://https://www.health.qld.gov.au/__data/assets/pdf_file/0025/444490/icare-pivc-guideline.pdf\",\"internal-pdf://https://www.mdpi.com/2076-0817/9/2/148\",\"internal-pdf://https://www.medrxiv.org/content/medrxiv/early/2020/01/30/2020.01.27.20018986.full.pdfhttp://orcid.org/0000-0003-4169-5665\",\"internal-pdf://https://academic.oup.com/inthealth/advance-article-abstract/doi/10.1093/inthealth/ihaa001/5732900\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32032682\",\"https://www.ncbi.nlm.nih.gov/pubmed/31981224\",\"internal-pdf://\",\"https://www.ncbi.nlm.nih.gov/pubmed/32050080\",\"internal-pdf://\",\"internal-pdf://Zhang et al. - Unknown - Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replication Structure-based design.pdfhttp://dx.doi.org/10.1101/2020.02.10.936898http://biorxiv.org/content/early/2020/02/10/2020.02.10.936898.abstract\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967327\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729071\",\"internal-pdf://\",\"internal-pdf://http://link.springer.com/10.1007/s00330-020-06731-x\",\"https://www.ncbi.nlm.nih.gov/pubmed/32031264\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729069\",\"internal-pdf://\",\"internal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan.pdfinternal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan(2).pdfhttps://www.medrxiv.org/content/10.1101/2020.01.28.20019299v3.full.pdf\",\"http://www.ncbi.nlm.nih.gov/pubmed/32093211https://www.mdpi.com/2077-0383/9/2/575\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/462\",\"https://www.mdpi.com/1999-4915/12/2/135\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32033064\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25699\",\"https://www.ncbi.nlm.nih.gov/pubmed/32030925\",\"https://www.ncbi.nlm.nih.gov/pubmed/31987001\",\"https://www.mdpi.com/2077-0383/9/2/330\",\"internal-pdf://https://www.mdpi.com/2218-273X/10/2/331\",\"internal-pdf://https://academic.oup.com/jid/advance-article/doi/10.1093/infdis/jiaa077/5739751\",\"internal-pdf://Zhao et al. - 2020 - Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 202.pdfhttps://www.ncbi.nlm.nih.gov/pubmed/32007643https://www.clinicalkey.com/#!/content/playContent/1-s2.0-S1201971220300539?returnurl=https:%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1201971220300539%3Fshowall%3Dtrue&referrer=https:%2F%2Fwww.ncbi.nlm.\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033513\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/498\",\"internal-pdf://Ki - Unknown - Epidemiologic characteristics of early cases with 2019-nCoV disease in Republ.pdfhttp://www.ncbi.nlm.nih.gov/pubmed/32035431http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020007\",\"https://www.ncbi.nlm.nih.gov/pubmed/32038520https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6992538/\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/596\",\"https://www.ncbi.nlm.nih.gov/pubmed/31943059\",\"https://actamedicaportuguesa.com/revista/index.php/amp/article/view/13547\",\"https://ui.adsabs.harvard.edu/abs/2020arXiv200200426Lhttp://arxiv.org/abs/2002.00426\",\"internal-pdf://https://www.mdpi.com/1999-4915/12/2/194\",\"https://www.cambridge.org/core/product/identifier/S0950268820000424/type/journal_article\",\"http://www.ncbi.nlm.nih.gov/pubmed/32035430\",\"https://www.biorxiv.org/content/biorxiv/early/2020/01/27/2020.01.26.920249.full.pdf\",\"internal-pdf://https://clintransmed.springeropen.com/articles/10.1186/s40169-020-00271-z\",\"internal-pdf://https://wellcomeopenresearch.org/articles/5-17/v1\",\"internal-pdf://https://www.tandfonline.com/action/journalInformation?journalCode=tveq20https://www.tandfonline.com/doi/full/10.1080/01652176.2020.1727993\",\"http://www.ncbi.nlm.nih.gov/pubmed/32023681\",\"https://www.ncbi.nlm.nih.gov/pubmed/32029004\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25707\",\"https://www.ncbi.nlm.nih.gov/pubmed/32013309\",\"internal-pdf://https://linkinghub.elsevier.com/retrieve/pii/S246804272030004X\",\"internal-pdf://https://jidc.org/index.php/journal/article/view/12425\",\"https://www.ncbi.nlm.nih.gov/pubmed/32017984\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967321\",\"internal-pdf://https://academic.oup.com/jtm/advance-article/doi/10.1093/jtm/taaa021/5735319\",\"https://www.tandfonline.com/doi/full/10.1080/20477724.2020.1725339\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033514\",\"internal-pdf://https://academic.oup.com/cid/advance-article/doi/10.1093/cid/ciaa149/5734265\",\"https://www.ncbi.nlm.nih.gov/pubmed/32007627\",\"internal-pdf://https://wellcomeopenresearch.org/articles/5-17/v1\",\"internal-pdf://https://www.medrxiv.org/content/medrxiv/early/2020/01/30/2020.01.27.20018986.full.pdfhttp://orcid.org/0000-0003-4169-5665\",\"https://www.tandfonline.com/doi/full/10.1080/20477724.2020.1725339\",\"https://www.ncbi.nlm.nih.gov/pubmed/31943059\",\"internal-pdf://\",\"https://www.ncbi.nlm.nih.gov/pubmed/31987001\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32032682\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967327\",\"http://www.ncbi.nlm.nih.gov/pubmed/32035430\",\"http://www.ncbi.nlm.nih.gov/pubmed/32093211https://www.mdpi.com/2077-0383/9/2/575\",\"internal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan.pdfinternal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan(2).pdfhttps://www.medrxiv.org/content/10.1101/2020.01.28.20019299v3.full.pdf\",\"https://actamedicaportuguesa.com/revista/index.php/amp/article/view/13547\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25699\",\"internal-pdf://https://www.mdpi.com/2218-273X/10/2/331\",\"https://www.mdpi.com/1999-4915/12/2/135\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033514\",\"https://www.cambridge.org/core/product/identifier/S0950268820000424/type/journal_article\",\"internal-pdf://\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967321\",\"https://www.ncbi.nlm.nih.gov/pubmed/32029004\",\"internal-pdf://\",\"internal-pdf://Ki - Unknown - Epidemiologic characteristics of early cases with 2019-nCoV disease in Republ.pdfhttp://www.ncbi.nlm.nih.gov/pubmed/32035431http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020007\",\"https://www.ncbi.nlm.nih.gov/pubmed/32030925\",\"https://www.ncbi.nlm.nih.gov/pubmed/32013309\",\"internal-pdf://http://doi.wiley.com/10.1002/jmv.25723\",\"https://www.ncbi.nlm.nih.gov/pubmed/31981224\",\"internal-pdf://https://academic.oup.com/inthealth/advance-article-abstract/doi/10.1093/inthealth/ihaa001/5732900\",\"https://www.ncbi.nlm.nih.gov/pubmed/32038520https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6992538/\",\"https://ui.adsabs.harvard.edu/abs/2020arXiv200200426Lhttp://arxiv.org/abs/2002.00426\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033513\",\"internal-pdf://https://academic.oup.com/jtm/advance-article/doi/10.1093/jtm/taaa021/5735319\",\"internal-pdf://https://www.tandfonline.com/action/journalInformation?journalCode=tveq20https://www.tandfonline.com/doi/full/10.1080/01652176.2020.1727993\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32033064\",\"https://www.mdpi.com/2077-0383/9/2/330\",\"internal-pdf://http://link.springer.com/10.1007/s00330-020-06731-x\",\"https://www.biorxiv.org/content/biorxiv/early/2020/01/27/2020.01.26.920249.full.pdf\",\"https://www.ncbi.nlm.nih.gov/pubmed/32007627\",\"internal-pdf://https://jidc.org/index.php/journal/article/view/12425\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/596\",\"internal-pdf://https://www.mdpi.com/1999-4915/12/2/194\",\"internal-pdf://https://www.mdpi.com/2076-0817/9/2/148\",\"internal-pdf://https://clintransmed.springeropen.com/articles/10.1186/s40169-020-00271-z\",\"https://pubmed.ncbi.nlm.nih.gov/32129843\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/462\",\"internal-pdf://https://linkinghub.elsevier.com/retrieve/pii/S246804272030004X\",\"https://www.ncbi.nlm.nih.gov/pubmed/32050080\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/498\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729069\",\"internal-pdf://https://academic.oup.com/cid/advance-article/doi/10.1093/cid/ciaa149/5734265\",\"internal-pdf://http://www.nature.com/articles/s41421-020-0148-0\",\"http://www.ncbi.nlm.nih.gov/pubmed/32023681\",\"https://www.ncbi.nlm.nih.gov/pubmed/32031264\",\"https://pubmed.ncbi.nlm.nih.gov/32150618\",\"https://www.ncbi.nlm.nih.gov/pubmed/32017984\",\"internal-pdf://https://academic.oup.com/jid/advance-article/doi/10.1093/infdis/jiaa077/5739751\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25707\",\"internal-pdf://Zhang et al. - Unknown - Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replication Structure-based design.pdfhttp://dx.doi.org/10.1101/2020.02.10.936898http://biorxiv.org/content/early/2020/02/10/2020.02.10.936898.abstract\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729071\",\"internal-pdf://Zhao et al. - 2020 - Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 202.pdfhttps://www.ncbi.nlm.nih.gov/pubmed/32007643https://www.clinicalkey.com/#!/content/playContent/1-s2.0-S1201971220300539?returnurl=https:%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1201971220300539%3Fshowall%3Dtrue&referrer=https:%2F%2Fwww.ncbi.nlm.\",\"internal-pdf://\",\"internal-pdf://https://wellcomeopenresearch.org/articles/5-17/v1\",\"internal-pdf://https://www.medrxiv.org/content/medrxiv/early/2020/01/30/2020.01.27.20018986.full.pdfhttp://orcid.org/0000-0003-4169-5665\",\"https://www.tandfonline.com/doi/full/10.1080/20477724.2020.1725339\",\"https://www.ncbi.nlm.nih.gov/pubmed/31943059\",\"internal-pdf://\",\"https://www.ncbi.nlm.nih.gov/pubmed/31987001\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32032682\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171869\",\"internal-pdf://https://f1000research.com/articles/9-129/v1\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967327\",\"http://www.ncbi.nlm.nih.gov/pubmed/32035430\",\"internal-pdf://http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020011\",\"http://www.ncbi.nlm.nih.gov/pubmed/32093211https://www.mdpi.com/2077-0383/9/2/575\",\"internal-pdf://https://f1000research.com/articles/9-121/v2\",\"http://www.ncbi.nlm.nih.gov/pubmed/32164081\",\"internal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan.pdfinternal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan(2).pdfhttps://www.medrxiv.org/content/10.1101/2020.01.28.20019299v3.full.pdf\",\"https://actamedicaportuguesa.com/revista/index.php/amp/article/view/13547\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25699\",\"internal-pdf://https://www.mdpi.com/2218-273X/10/2/331\",\"https://www.mdpi.com/1999-4915/12/2/135\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033514\",\"https://www.cambridge.org/core/product/identifier/S0950268820000424/type/journal_article\",\"internal-pdf://https://f1000research.com/articles/9-145/v1\",\"internal-pdf://\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967321\",\"https://www.ncbi.nlm.nih.gov/pubmed/32029004\",\"internal-pdf://\",\"internal-pdf://Ki - Unknown - Epidemiologic characteristics of early cases with 2019-nCoV disease in Republ.pdfhttp://www.ncbi.nlm.nih.gov/pubmed/32035431http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020007\",\"https://www.ncbi.nlm.nih.gov/pubmed/32030925\",\"https://www.ncbi.nlm.nih.gov/pubmed/32013309\",\"internal-pdf://http://doi.wiley.com/10.1002/jmv.25723\",\"https://en.wikipedia.org/wiki/2020_Hubei_lockdownshttp://www.ncbi.nlm.nih.gov/pubmed/32167146\",\"https://www.ncbi.nlm.nih.gov/pubmed/31981224\",\"http://www.ncbi.nlm.nih.gov/pubmed/32164080\",\"internal-pdf://https://academic.oup.com/inthealth/advance-article-abstract/doi/10.1093/inthealth/ihaa001/5732900\",\"https://www.ncbi.nlm.nih.gov/pubmed/32038520https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6992538/\",\"https://ui.adsabs.harvard.edu/abs/2020arXiv200200426Lhttp://arxiv.org/abs/2002.00426\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033513\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171866\",\"internal-pdf://https://academic.oup.com/jtm/advance-article/doi/10.1093/jtm/taaa021/5735319\",\"internal-pdf://https://www.tandfonline.com/action/journalInformation?journalCode=tveq20https://www.tandfonline.com/doi/full/10.1080/01652176.2020.1727993\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32033064\",\"https://www.mdpi.com/2077-0383/9/2/330\",\"internal-pdf://\",\"internal-pdf://http://link.springer.com/10.1007/s00330-020-06731-x\",\"https://www.biorxiv.org/content/biorxiv/early/2020/01/27/2020.01.26.920249.full.pdf\",\"https://www.ncbi.nlm.nih.gov/pubmed/32007627\",\"internal-pdf://https://jidc.org/index.php/journal/article/view/12425\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/596\",\"internal-pdf://https://www.mdpi.com/1999-4915/12/2/194\",\"internal-pdf://https://www.mdpi.com/2076-0817/9/2/148\",\"internal-pdf://https://clintransmed.springeropen.com/articles/10.1186/s40169-020-00271-z\",\"https://pubmed.ncbi.nlm.nih.gov/32129843\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171192\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/462\",\"internal-pdf://https://linkinghub.elsevier.com/retrieve/pii/S246804272030004X\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171948\",\"https://www.ncbi.nlm.nih.gov/pubmed/32050080\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/498\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729069\",\"internal-pdf://https://academic.oup.com/cid/advance-article/doi/10.1093/cid/ciaa149/5734265\",\"internal-pdf://\",\"internal-pdf://http://www.nature.com/articles/s41421-020-0148-0\",\"http://www.ncbi.nlm.nih.gov/pubmed/32023681\",\"https://www.ncbi.nlm.nih.gov/pubmed/32031264\",\"https://pubmed.ncbi.nlm.nih.gov/32150618\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171867\",\"https://www.ncbi.nlm.nih.gov/pubmed/32017984\",\"internal-pdf://https://academic.oup.com/jid/advance-article/doi/10.1093/infdis/jiaa077/5739751\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25707\",\"internal-pdf://Zhang et al. - Unknown - Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replication Structure-based design.pdfhttp://dx.doi.org/10.1101/2020.02.10.936898http://biorxiv.org/content/early/2020/02/10/2020.02.10.936898.abstract\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729071\",\"internal-pdf://Zhao et al. - 2020 - Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 202.pdfhttps://www.ncbi.nlm.nih.gov/pubmed/32007643https://www.clinicalkey.com/#!/content/playContent/1-s2.0-S1201971220300539?returnurl=https:%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1201971220300539%3Fshowall%3Dtrue&referrer=https:%2F%2Fwww.ncbi.nlm.\",\"internal-pdf://\",\"internal-pdf://www.preprints.org\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171951\",\"internal-pdf://https://wellcomeopenresearch.org/articles/5-17/v1\",\"internal-pdf://https://www.medrxiv.org/content/medrxiv/early/2020/01/30/2020.01.27.20018986.full.pdfhttp://orcid.org/0000-0003-4169-5665\",\"https://www.tandfonline.com/doi/full/10.1080/20477724.2020.1725339\",\"internal-pdf://\",\"https://www.ncbi.nlm.nih.gov/pubmed/31943059\",\"internal-pdf://\",\"https://www.ncbi.nlm.nih.gov/pubmed/31987001\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32032682\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171869\",\"internal-pdf://https://f1000research.com/articles/9-129/v1\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967327\",\"http://www.ncbi.nlm.nih.gov/pubmed/32035430\",\"internal-pdf://http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020011\",\"http://www.ncbi.nlm.nih.gov/pubmed/32093211https://www.mdpi.com/2077-0383/9/2/575\",\"internal-pdf://https://f1000research.com/articles/9-121/v2\",\"http://www.ncbi.nlm.nih.gov/pubmed/32164081\",\"internal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan.pdfinternal-pdf://Du et al. - 2020 - \\u200bRisk of 2019 novel coronavirus importations throughout China prior to the Wuhan(2).pdfhttps://www.medrxiv.org/content/10.1101/2020.01.28.20019299v3.full.pdf\",\"https://actamedicaportuguesa.com/revista/index.php/amp/article/view/13547\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25699\",\"internal-pdf://https://www.mdpi.com/2218-273X/10/2/331\",\"https://www.mdpi.com/1999-4915/12/2/135\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033514\",\"https://www.cambridge.org/core/product/identifier/S0950268820000424/type/journal_article\",\"internal-pdf://https://f1000research.com/articles/9-145/v1\",\"internal-pdf://\",\"https://www.ncbi.nlm.nih.gov/pubmed/31967321\",\"https://www.ncbi.nlm.nih.gov/pubmed/32029004\",\"internal-pdf://\",\"internal-pdf://Ki - Unknown - Epidemiologic characteristics of early cases with 2019-nCoV disease in Republ.pdfhttp://www.ncbi.nlm.nih.gov/pubmed/32035431http://e-epih.org/journal/view.php?doi=10.4178/epih.e2020007\",\"https://www.ncbi.nlm.nih.gov/pubmed/32030925\",\"https://www.ncbi.nlm.nih.gov/pubmed/32013309\",\"internal-pdf://http://doi.wiley.com/10.1002/jmv.25723\",\"internal-pdf://https://academic.oup.com/jtm/advance-article-abstract/doi/10.1093/jtm/taaa037/5808003\",\"https://en.wikipedia.org/wiki/2020_Hubei_lockdownshttp://www.ncbi.nlm.nih.gov/pubmed/32167146\",\"https://www.ncbi.nlm.nih.gov/pubmed/31981224\",\"http://www.ncbi.nlm.nih.gov/pubmed/32164080\",\"internal-pdf://https://academic.oup.com/inthealth/advance-article-abstract/doi/10.1093/inthealth/ihaa001/5732900\",\"https://www.ncbi.nlm.nih.gov/pubmed/32038520https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6992538/\",\"https://ui.adsabs.harvard.edu/abs/2020arXiv200200426Lhttp://arxiv.org/abs/2002.00426\",\"https://www.ncbi.nlm.nih.gov/pubmed/32033513\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171866\",\"internal-pdf://https://academic.oup.com/jtm/advance-article/doi/10.1093/jtm/taaa021/5735319\",\"internal-pdf://https://www.tandfonline.com/action/journalInformation?journalCode=tveq20https://www.tandfonline.com/doi/full/10.1080/01652176.2020.1727993\",\"internal-pdf://\",\"internal-pdf://https://www.ncbi.nlm.nih.gov/pubmed/32033064\",\"https://www.mdpi.com/2077-0383/9/2/330\",\"internal-pdf://\",\"internal-pdf://http://link.springer.com/10.1007/s00330-020-06731-x\",\"https://www.biorxiv.org/content/biorxiv/early/2020/01/27/2020.01.26.920249.full.pdf\",\"https://www.ncbi.nlm.nih.gov/pubmed/32007627\",\"internal-pdf://https://jidc.org/index.php/journal/article/view/12425\",\"internal-pdf://https://academic.oup.com/jtm/advance-article-abstract/doi/10.1093/jtm/taaa030/5766334\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/596\",\"internal-pdf://https://www.mdpi.com/1999-4915/12/2/194\",\"internal-pdf://https://www.mdpi.com/2076-0817/9/2/148\",\"internal-pdf://https://clintransmed.springeropen.com/articles/10.1186/s40169-020-00271-z\",\"https://pubmed.ncbi.nlm.nih.gov/32129843\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171192\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/462\",\"internal-pdf://https://linkinghub.elsevier.com/retrieve/pii/S246804272030004X\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171948\",\"https://www.ncbi.nlm.nih.gov/pubmed/32050080\",\"internal-pdf://https://www.mdpi.com/2077-0383/9/2/498\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729069\",\"internal-pdf://https://academic.oup.com/cid/advance-article/doi/10.1093/cid/ciaa149/5734265\",\"internal-pdf://\",\"internal-pdf://http://www.nature.com/articles/s41421-020-0148-0\",\"http://www.ncbi.nlm.nih.gov/pubmed/32023681\",\"https://www.ncbi.nlm.nih.gov/pubmed/32031264\",\"https://pubmed.ncbi.nlm.nih.gov/32150618\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171867\",\"https://www.ncbi.nlm.nih.gov/pubmed/32017984\",\"internal-pdf://https://academic.oup.com/jid/advance-article/doi/10.1093/infdis/jiaa077/5739751\",\"https://onlinelibrary.wiley.com/doi/abs/10.1002/jmv.25707\",\"internal-pdf://Zhang et al. - Unknown - Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replication Structure-based design.pdfhttp://dx.doi.org/10.1101/2020.02.10.936898http://biorxiv.org/content/early/2020/02/10/2020.02.10.936898.abstract\",\"internal-pdf://https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1729071\",\"internal-pdf://Zhao et al. - 2020 - Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 202.pdfhttps://www.ncbi.nlm.nih.gov/pubmed/32007643https://www.clinicalkey.com/#!/content/playContent/1-s2.0-S1201971220300539?returnurl=https:%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1201971220300539%3Fshowall%3Dtrue&referrer=https:%2F%2Fwww.ncbi.nlm.\",\"internal-pdf://\",\"internal-pdf://www.preprints.org\",\"http://www.ncbi.nlm.nih.gov/pubmed/32171951\"],\"titles\":[\"The incubation period of 2019-nCoV infections among travellers from Wuhan, ChinamedRxiv\",\"Pneumonia of Unknown Etiology in Wuhan, China: Potential for International Spread Via Commercial Air TravelJournal of travel medicine\",\"Genomic characterization of the 2019 novel human-pathogenic coronavirus isolated from a patient with atypical pneumonia after visiting WuhanEmerging microbes & infections\",\"Pathogenicity and Transmissibility of 2019-nCoV-A Quick Overview and Comparison with Other Emerging VirusesMicrobes and infection\",\"Emerging coronaviruses: genome structure, replication, and pathogenesisJournal of medical virology\",\"[Recommendations on extracorporeal life support for critically ill patients with novel coronavirus pneumonia].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"The 2019 Novel Coronavirus (2019-nCoV): Novel Virus, Old ChallengesActa M\\u00e9dica Portuguesa\",\"The first two cases of 2019\\u2010nCoV in Italy: where they come from?Journal of Medical Virology\",\"Return of the Coronavirus: 2019-nCoVViruses\",\"Expert consensus for bronchoscopy during the epidemic of 2019 Novel Coronavirus infection (Trial version)Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Homologous recombination within the spike glycoprotein of the newly identified coronavirus may boost cross-species transmission from snake to humanJournal of medical virology\",\"A rapid advice guideline for the diagnosis and treatment of 2019 novel coronavirus (2019-nCoV) infected pneumonia (standard version)Military Medical Research\",\"Epidemiologic characteristics of early cases with 2019 novel coronavirus (2019-nCoV) disease in Republic of KoreaEpidemiology and Health\",\"Epidemiologic characteristics of early cases with 2019 novel coronavirus (2019-nCoV) disease in Republic of KoreaEpidemiology and Health\",\"The First Case of 2019 Novel Coronavirus Pneumonia Imported into Korea from Wuhan, China: Implication for Infection Prevention and Control MeasuresJournal of Korean medical science\",\"Mast cells contribute to coronavirus-induced inflammation: new anti-inflammatory strategyJournal of biological regulators and homeostatic agents\",\"Coronavirus Infections and Immune ResponsesJournal of medical virology\",\"OUP accepted manuscriptInternational Health\",\"Regulation of the ER Stress Response by the Ion Channel Activity of the Infectious Bronchitis Coronavirus Envelope Protein Modulates Virion Release, Apoptosis, Viral Fitness, and PathogenesisFrontiers in microbiology\",\"A Simple Prediction Model for the Development Trend of 2019-nCov Epidemics Based on Medical ObservationsarXiv preprint arXiv:2002.00426\",\"Interpretation of \\\"Guidelines for the Diagnosis and Treatment of Novel Coronavirus (2019-nCoV) Infection by the National Health Commission (Trial Version 5)\\\"Zhonghua yi xue za zhi\",\"Emerging novel Coronavirus (2019-nCoV) - Current scenario, evolutionary perspective based on genome analysis and recent developmentsVeterinary Quarterly\",\"The Rate of Underascertainment of Novel Coronavirus (2019-nCoV) Infection: Estimation Using Japanese Passengers Data on Evacuation FlightsJournal of clinical medicine\",\"The Extent of Transmission of Novel Coronavirus in Wuhan, China, 2020Journal of Clinical Medicine\",\"Full-genome evolutionary analysis of the novel corona virus (2019-nCoV) rejects the hypothesis of emergence as a result of a recent recombination eventInfection, Genetics and Evolution\",\"Novel coronavirus: From discovery to clinical diagnosticsInfection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"[Diagnosis and clinical management of 2019 novel coronavirus infection: an operational recommendation of Peking Union Medical College Hospital (V2.0)].Zhonghua nei ke za zhi\",\"Evolving status of the 2019 novel coronavirus Infection: proposal of conventional serologic assays for disease diagnosis and infection monitoring [Commentary/Review]Journal of medical virology\",\"Measures for diagnosing and treating infections by a novel coronavirus responsible for a pneumonia outbreak originating in Wuhan, ChinaMicrobes and infection\",\"Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replicationbioRxiv\",\"Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 2020: A data-driven analysis in the early phase of the outbreakInternational journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Preliminary prediction of the basic reproduction number of the Wuhan novel coronavirus 2019-nCoVarXiv preprint arXiv:2001.10530\",\"The incubation period of 2019-nCoV infections among travellers from Wuhan, ChinamedRxiv\",\"The global spread of 2019-nCoV: a molecular evolutionary analysisPathogens and Global Health\",\"Pneumonia of Unknown Etiology in Wuhan, China: Potential for International Spread Via Commercial Air TravelJournal of travel medicine\",\"Genomic characterization of the 2019 novel human-pathogenic coronavirus isolated from a patient with atypical pneumonia after visiting WuhanEmerging microbes & infections\",\"Pathogenicity and Transmissibility of 2019-nCoV-A Quick Overview and Comparison with Other Emerging VirusesMicrobes and infection\",\"Emerging coronaviruses: genome structure, replication, and pathogenesisJournal of medical virology\",\"[Recommendations on extracorporeal life support for critically ill patients with novel coronavirus pneumonia].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"The 2019 Novel Coronavirus (2019-nCoV): Novel Virus, Old ChallengesActa M\\u00e9dica Portuguesa\",\"The first two cases of 2019\\u2010nCoV in Italy: where they come from?Journal of Medical Virology\",\"Return of the Coronavirus: 2019-nCoVViruses\",\"Expert consensus for bronchoscopy during the epidemic of 2019 Novel Coronavirus infection (Trial version)Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Homologous recombination within the spike glycoprotein of the newly identified coronavirus may boost cross-species transmission from snake to humanJournal of medical virology\",\"A rapid advice guideline for the diagnosis and treatment of 2019 novel coronavirus (2019-nCoV) infected pneumonia (standard version)Military Medical Research\",\"Epidemiologic characteristics of early cases with 2019 novel coronavirus (2019-nCoV) disease in Republic of KoreaEpidemiology and Health\",\"Epidemiologic characteristics of early cases with 2019 novel coronavirus (2019-nCoV) disease in Republic of KoreaEpidemiology and Health\",\"The First Case of 2019 Novel Coronavirus Pneumonia Imported into Korea from Wuhan, China: Implication for Infection Prevention and Control MeasuresJournal of Korean medical science\",\"Mast cells contribute to coronavirus-induced inflammation: new anti-inflammatory strategyJournal of biological regulators and homeostatic agents\",\"Coronavirus Infections and Immune ResponsesJournal of medical virology\",\"OUP accepted manuscriptInternational Health\",\"Regulation of the ER Stress Response by the Ion Channel Activity of the Infectious Bronchitis Coronavirus Envelope Protein Modulates Virion Release, Apoptosis, Viral Fitness, and PathogenesisFrontiers in microbiology\",\"A Simple Prediction Model for the Development Trend of 2019-nCov Epidemics Based on Medical ObservationsarXiv preprint arXiv:2002.00426\",\"Interpretation of \\\"Guidelines for the Diagnosis and Treatment of Novel Coronavirus (2019-nCoV) Infection by the National Health Commission (Trial Version 5)\\\"Zhonghua yi xue za zhi\",\"The reproductive number of COVID-19 is higher compared to SARS coronavirusJournal of Travel Medicine\",\"Emerging novel Coronavirus (2019-nCoV) - Current scenario, evolutionary perspective based on genome analysis and recent developmentsVeterinary Quarterly\",\"The Rate of Underascertainment of Novel Coronavirus (2019-nCoV) Infection: Estimation Using Japanese Passengers Data on Evacuation FlightsJournal of clinical medicine\",\"The Extent of Transmission of Novel Coronavirus in Wuhan, China, 2020Journal of Clinical Medicine\",\"Initial CT findings and temporal changes in patients with the novel coronavirus pneumonia (2019-nCoV): a study of 63 patients in Wuhan, ChinaEuropean Radiology\",\"Full-genome evolutionary analysis of the novel corona virus (2019-nCoV) rejects the hypothesis of emergence as a result of a recent recombination eventInfection, Genetics and Evolution\",\"Novel coronavirus: From discovery to clinical diagnosticsInfection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"Potential Maternal and Infant Outcomes from (Wuhan) Coronavirus 2019-nCoV Infecting Pregnant Women: Lessons from SARS, MERS, and Other Human Coronavirus InfectionsViruses\",\"Estimation of the Transmission Risk of the 2019-nCoV and Its Implication for Public Health InterventionsJournal of Clinical Medicine\",\"COVID-19, Australia: Epidemiology Report 2 (Reporting week ending 19:00 AEDT 8 February 2020)Communicable diseases intelligence (2018)\",\"Novel Coronavirus Outbreak in Wuhan, China, 2020: Intense Surveillance Is Vital for Preventing Sustained Transmission in New LocationsJournal of Clinical Medicine\",\"Consistent detection of 2019 novel coronavirus in salivaClinical Infectious Diseases\",\"[Diagnosis and clinical management of 2019 novel coronavirus infection: an operational recommendation of Peking Union Medical College Hospital (V2.0)].Zhonghua nei ke za zhi\",\"Evolving status of the 2019 novel coronavirus Infection: proposal of conventional serologic assays for disease diagnosis and infection monitoring [Commentary/Review]Journal of medical virology\",\"Measures for diagnosing and treating infections by a novel coronavirus responsible for a pneumonia outbreak originating in Wuhan, ChinaMicrobes and infection\",\"Potential Interventions for Novel Coronavirus in China: A Systemic ReviewJournal of Medical Virology\",\"Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replicationbioRxiv\",\"Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 2020: A data-driven analysis in the early phase of the outbreakInternational journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Preliminary prediction of the basic reproduction number of the Wuhan novel coronavirus 2019-nCoVarXiv preprint arXiv:2001.10530\",\"The transmissibility of novel Coronavirus in the early stages of the 2019-20 outbreak in Wuhan: Exploring initial point-source exposure sizes and durations using scenario analysisWellcome Open Research\",\"The incubation period of 2019-nCoV infections among travellers from Wuhan, ChinamedRxiv\",\"The global spread of 2019-nCoV: a molecular evolutionary analysisPathogens and Global Health\",\"Pneumonia of Unknown Etiology in Wuhan, China: Potential for International Spread Via Commercial Air TravelJournal of travel medicine\",\"Risk Assessment of Novel Coronavirus COVID-19 Outbreaks Outside ChinaJournal of Clinical Medicine 2020, Vol. 9, Page 571\",\"Genomic characterization of the 2019 novel human-pathogenic coronavirus isolated from a patient with atypical pneumonia after visiting WuhanEmerging microbes & infections\",\"Pathogenicity and Transmissibility of 2019-nCoV-A Quick Overview and Comparison with Other Emerging VirusesMicrobes and infection\",\"Emerging coronaviruses: genome structure, replication, and pathogenesisJournal of medical virology\",\"[Recommendations on extracorporeal life support for critically ill patients with novel coronavirus pneumonia].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Characteristics of and Public Health Responses to the Coronavirus Disease 2019 Outbreak in ChinaJournal of Clinical Medicine\",\"Risk of 2019 novel coronavirus importations throughout China prior to the Wuhan quarantinemedRxiv\",\"The 2019 Novel Coronavirus (2019-nCoV): Novel Virus, Old ChallengesActa M\\u00e9dica Portuguesa\",\"The first two cases of 2019\\u2010nCoV in Italy: where they come from?Journal of Medical Virology\",\"Rigidity of the Outer Shell Predicted by a Protein Intrinsic Disorder Model Sheds Light on the COVID-19 (Wuhan-2019-nCoV) InfectivityBiomolecules\",\"Return of the Coronavirus: 2019-nCoVViruses\",\"Expert consensus for bronchoscopy during the epidemic of 2019 Novel Coronavirus infection (Trial version)Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Homologous recombination within the spike glycoprotein of the newly identified coronavirus may boost cross-species transmission from snake to humanJournal of medical virology\",\"A rapid advice guideline for the diagnosis and treatment of 2019 novel coronavirus (2019-nCoV) infected pneumonia (standard version)Military Medical Research\",\"Real-Time Estimation of the Risk of Death from Novel Coronavirus (COVID-19) Infection: Inference Using Exported CasesJournal of Clinical Medicine 2020, Vol. 9, Page 523\",\"Epidemiologic characteristics of early cases with 2019 novel coronavirus (2019-nCoV) disease in Republic of KoreaEpidemiology and Health\",\"The First Case of 2019 Novel Coronavirus Pneumonia Imported into Korea from Wuhan, China: Implication for Infection Prevention and Control MeasuresJournal of Korean medical science\",\"Mast cells contribute to coronavirus-induced inflammation: new anti-inflammatory strategyJournal of biological regulators and homeostatic agents\",\"Coronavirus Infections and Immune ResponsesJournal of medical virology\",\"OUP accepted manuscriptInternational Health\",\"Regulation of the ER Stress Response by the Ion Channel Activity of the Infectious Bronchitis Coronavirus Envelope Protein Modulates Virion Release, Apoptosis, Viral Fitness, and PathogenesisFrontiers in microbiology\",\"A Simple Prediction Model for the Development Trend of 2019-nCov Epidemics Based on Medical ObservationsarXiv preprint arXiv:2002.00426\",\"Interpretation of \\\"Guidelines for the Diagnosis and Treatment of Novel Coronavirus (2019-nCoV) Infection by the National Health Commission (Trial Version 5)\\\"Zhonghua yi xue za zhi\",\"The reproductive number of COVID-19 is higher compared to SARS coronavirusJournal of Travel Medicine\",\"Emerging novel Coronavirus (2019-nCoV) - Current scenario, evolutionary perspective based on genome analysis and recent developmentsVeterinary Quarterly\",\"The Rate of Underascertainment of Novel Coronavirus (2019-nCoV) Infection: Estimation Using Japanese Passengers Data on Evacuation FlightsJournal of clinical medicine\",\"The Extent of Transmission of Novel Coronavirus in Wuhan, China, 2020Journal of Clinical Medicine\",\"Initial CT findings and temporal changes in patients with the novel coronavirus pneumonia (2019-nCoV): a study of 63 patients in Wuhan, ChinaEuropean Radiology\",\"Full-genome evolutionary analysis of the novel corona virus (2019-nCoV) rejects the hypothesis of emergence as a result of a recent recombination eventInfection, Genetics and Evolution\",\"Novel coronavirus: From discovery to clinical diagnosticsInfection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"2019-nCoV (Wuhan virus), a novel Coronavirus: Human-to-human transmission, travel-related cases, and vaccine readinessJournal of Infection in Developing Countries\",\"Potential Maternal and Infant Outcomes from (Wuhan) Coronavirus 2019-nCoV Infecting Pregnant Women: Lessons from SARS, MERS, and Other Human Coronavirus InfectionsViruses\",\"2019 novel coronavirus of pneumonia in Wuhan, China: emerging attack and management strategiesClinical and Translational Medicine\",\"Estimation of the Transmission Risk of the 2019-nCoV and Its Implication for Public Health InterventionsJournal of Clinical Medicine\",\"COVID-19, Australia: Epidemiology Report 2 (Reporting week ending 19:00 AEDT 8 February 2020)Communicable diseases intelligence (2018)\",\"Novel Coronavirus Outbreak in Wuhan, China, 2020: Intense Surveillance Is Vital for Preventing Sustained Transmission in New LocationsJournal of Clinical Medicine\",\"Potent binding of 2019 novel coronavirus spike protein by a SARS coronavirus-specific human monoclonal antibodyEmerging Microbes & Infections\",\"Consistent detection of 2019 novel coronavirus in salivaClinical Infectious Diseases\",\"[Diagnosis and clinical management of 2019 novel coronavirus infection: an operational recommendation of Peking Union Medical College Hospital (V2.0)].Zhonghua nei ke za zhi\",\"Evolving status of the 2019 novel coronavirus Infection: proposal of conventional serologic assays for disease diagnosis and infection monitoring [Commentary/Review]Journal of medical virology\",\"Measures for diagnosing and treating infections by a novel coronavirus responsible for a pneumonia outbreak originating in Wuhan, ChinaMicrobes and infection\",\"A familial cluster of infection associated with the 2019 novel coronavirus indicating potential person-to-person transmission during the incubation periodThe Journal of Infectious Diseases\",\"Potential Interventions for Novel Coronavirus in China: A Systemic ReviewJournal of Medical Virology\",\"Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replicationbioRxiv\",\"Molecular and serological investigation of 2019-nCoV infected patients: implication of multiple shedding routesEmerging Microbes & Infections\",\"Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 2020: A data-driven analysis in the early phase of the outbreakInternational journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Preliminary prediction of the basic reproduction number of the Wuhan novel coronavirus 2019-nCoVarXiv preprint arXiv:2001.10530\",\"Peripheral Intravenous Catheter Protection :British Journal of Nursing\",\"Peripheral Intravenous Catheter Protection :British Journal of Nursing\",\"Emergence of Novel Coronavirus 2019-nCoV: Need for Rapid Vaccine and Biologics DevelopmentPathogens\",\"The incubation period of 2019-nCoV infections among travellers from Wuhan, ChinamedRxiv\",\"OUP accepted manuscriptInternational Health\",\"Pathogenicity and Transmissibility of 2019-nCoV-A Quick Overview and Comparison with Other Emerging VirusesMicrobes and infection\",\"Coronavirus Infections and Immune ResponsesJournal of medical virology\",\"Preliminary prediction of the basic reproduction number of the Wuhan novel coronavirus 2019-nCoVarXiv preprint arXiv:2001.10530\",\"COVID-19, Australia: Epidemiology Report 2 (Reporting week ending 19:00 AEDT 8 February 2020)Communicable diseases intelligence (2018)\",\"A Simulation on Potential Secondary Spread of Novel Coronavirus in an Exported Country Using a Stochastic Epidemic SEIR ModelPreprints (www.preprints.org)\",\"Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replicationbioRxiv\",\"Emerging coronaviruses: genome structure, replication, and pathogenesisJournal of medical virology\",\"Molecular and serological investigation of 2019-nCoV infected patients: implication of multiple shedding routesEmerging Microbes & Infections\",\"Risk Assessment of Novel Coronavirus COVID-19 Outbreaks Outside ChinaJournal of Clinical Medicine 2020, Vol. 9, Page 571\",\"Initial CT findings and temporal changes in patients with the novel coronavirus pneumonia (2019-nCoV): a study of 63 patients in Wuhan, ChinaEuropean Radiology\",\"Evolving status of the 2019 novel coronavirus Infection: proposal of conventional serologic assays for disease diagnosis and infection monitoring [Commentary/Review]Journal of medical virology\",\"Potent binding of 2019 novel coronavirus spike protein by a SARS coronavirus-specific human monoclonal antibodyEmerging Microbes & Infections\",\"Real-Time Estimation of the Risk of Death from Novel Coronavirus (COVID-19) Infection: Inference Using Exported CasesJournal of Clinical Medicine 2020, Vol. 9, Page 523\",\"Risk of 2019 novel coronavirus importations throughout China prior to the Wuhan quarantinemedRxiv\",\"Characteristics of and Public Health Responses to the Coronavirus Disease 2019 Outbreak in ChinaJournal of Clinical Medicine\",\"Estimation of the Transmission Risk of the 2019-nCoV and Its Implication for Public Health InterventionsJournal of Clinical Medicine\",\"Return of the Coronavirus: 2019-nCoVViruses\",\"The Rate of Underascertainment of Novel Coronavirus (2019-nCoV) Infection: Estimation Using Japanese Passengers Data on Evacuation FlightsJournal of clinical medicine\",\"The first two cases of 2019\\u2010nCoV in Italy: where they come from?Journal of Medical Virology\",\"The First Case of 2019 Novel Coronavirus Pneumonia Imported into Korea from Wuhan, China: Implication for Infection Prevention and Control MeasuresJournal of Korean medical science\",\"Genomic characterization of the 2019 novel human-pathogenic coronavirus isolated from a patient with atypical pneumonia after visiting WuhanEmerging microbes & infections\",\"The Extent of Transmission of Novel Coronavirus in Wuhan, China, 2020Journal of Clinical Medicine\",\"Rigidity of the Outer Shell Predicted by a Protein Intrinsic Disorder Model Sheds Light on the COVID-19 (Wuhan-2019-nCoV) InfectivityBiomolecules\",\"A familial cluster of infection associated with the 2019 novel coronavirus indicating potential person-to-person transmission during the incubation periodThe Journal of Infectious Diseases\",\"Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 2020: A data-driven analysis in the early phase of the outbreakInternational journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Interpretation of \\\"Guidelines for the Diagnosis and Treatment of Novel Coronavirus (2019-nCoV) Infection by the National Health Commission (Trial Version 5)\\\"Zhonghua yi xue za zhi\",\"Novel Coronavirus Outbreak in Wuhan, China, 2020: Intense Surveillance Is Vital for Preventing Sustained Transmission in New LocationsJournal of Clinical Medicine\",\"Epidemiologic characteristics of early cases with 2019 novel coronavirus (2019-nCoV) disease in Republic of KoreaEpidemiology and Health\",\"Regulation of the ER Stress Response by the Ion Channel Activity of the Infectious Bronchitis Coronavirus Envelope Protein Modulates Virion Release, Apoptosis, Viral Fitness, and PathogenesisFrontiers in microbiology\",\"Short-term Forecasts of the COVID-19 Epidemic in Guangdong and Zhejiang, China: February 13\\u201323, 2020Journal of Clinical Medicine\",\"Pneumonia of Unknown Etiology in Wuhan, China: Potential for International Spread Via Commercial Air TravelJournal of travel medicine\",\"The 2019 Novel Coronavirus (2019-nCoV): Novel Virus, Old ChallengesActa M\\u00e9dica Portuguesa\",\"A Simple Prediction Model for the Development Trend of 2019-nCov Epidemics Based on Medical ObservationsarXiv preprint arXiv:2002.00426\",\"Potential Maternal and Infant Outcomes from (Wuhan) Coronavirus 2019-nCoV Infecting Pregnant Women: Lessons from SARS, MERS, and Other Human Coronavirus InfectionsViruses\",\"Passengers' destinations from China: low risk of Novel Coronavirus (2019-nCoV) transmission into Africa and South AmericaEpidemiology and Infection\",\"[Recommendations on extracorporeal life support for critically ill patients with novel coronavirus pneumonia].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Full-genome evolutionary analysis of the novel corona virus (2019-nCoV) rejects the hypothesis of emergence as a result of a recent recombination eventInfection, Genetics and Evolution\",\"2019 novel coronavirus of pneumonia in Wuhan, China: emerging attack and management strategiesClinical and Translational Medicine\",\"The transmissibility of novel Coronavirus in the early stages of the 2019-20 outbreak in Wuhan: Exploring initial point-source exposure sizes and durations using scenario analysisWellcome Open Research\",\"Emerging novel Coronavirus (2019-nCoV) - Current scenario, evolutionary perspective based on genome analysis and recent developmentsVeterinary Quarterly\",\"[Diagnosis and clinical management of 2019 novel coronavirus infection: an operational recommendation of Peking Union Medical College Hospital (V2.0)].Zhonghua nei ke za zhi\",\"A rapid advice guideline for the diagnosis and treatment of 2019 novel coronavirus (2019-nCoV) infected pneumonia (standard version)Military Medical Research\",\"Potential Interventions for Novel Coronavirus in China: A Systemic ReviewJournal of Medical Virology\",\"Mast cells contribute to coronavirus-induced inflammation: new anti-inflammatory strategyJournal of biological regulators and homeostatic agents\",\"An updated estimation of the risk of transmission of the novel coronavirus (2019-nCov)Infectious Disease Modelling\",\"2019-nCoV (Wuhan virus), a novel Coronavirus: Human-to-human transmission, travel-related cases, and vaccine readinessJournal of Infection in Developing Countries\",\"Measures for diagnosing and treating infections by a novel coronavirus responsible for a pneumonia outbreak originating in Wuhan, ChinaMicrobes and infection\",\"Homologous recombination within the spike glycoprotein of the newly identified coronavirus may boost cross-species transmission from snake to humanJournal of medical virology\",\"The reproductive number of COVID-19 is higher compared to SARS coronavirusJournal of Travel Medicine\",\"The global spread of 2019-nCoV: a molecular evolutionary analysisPathogens and Global Health\",\"Expert consensus for bronchoscopy during the epidemic of 2019 Novel Coronavirus infection (Trial version)Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Consistent detection of 2019 novel coronavirus in salivaClinical Infectious Diseases\",\"Novel coronavirus: From discovery to clinical diagnosticsInfection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"The transmissibility of novel Coronavirus in the early stages of the 2019-20 outbreak in Wuhan: Exploring initial point-source exposure sizes and durations using scenario analysisWellcome Open Research\",\"The incubation period of 2019-nCoV infections among travellers from Wuhan, ChinamedRxiv\",\"The global spread of 2019-nCoV: a molecular evolutionary analysisPathogens and Global Health\",\"Pneumonia of Unknown Etiology in Wuhan, China: Potential for International Spread Via Commercial Air TravelJournal of travel medicine\",\"Risk Assessment of Novel Coronavirus COVID-19 Outbreaks Outside ChinaJournal of Clinical Medicine 2020, Vol. 9, Page 571\",\"Genomic characterization of the 2019 novel human-pathogenic coronavirus isolated from a patient with atypical pneumonia after visiting WuhanEmerging microbes & infections\",\"Pathogenicity and Transmissibility of 2019-nCoV-A Quick Overview and Comparison with Other Emerging VirusesMicrobes and infection\",\"Emerging coronaviruses: genome structure, replication, and pathogenesisJournal of medical virology\",\"[Recommendations on extracorporeal life support for critically ill patients with novel coronavirus pneumonia].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Characteristics of and Public Health Responses to the Coronavirus Disease 2019 Outbreak in ChinaJournal of Clinical Medicine\",\"Risk of 2019 novel coronavirus importations throughout China prior to the Wuhan quarantinemedRxiv\",\"The 2019 Novel Coronavirus (2019-nCoV): Novel Virus, Old ChallengesActa M\\u00e9dica Portuguesa\",\"The first two cases of 2019\\u2010nCoV in Italy: where they come from?Journal of Medical Virology\",\"Rigidity of the Outer Shell Predicted by a Protein Intrinsic Disorder Model Sheds Light on the COVID-19 (Wuhan-2019-nCoV) InfectivityBiomolecules\",\"Return of the Coronavirus: 2019-nCoVViruses\",\"Expert consensus for bronchoscopy during the epidemic of 2019 Novel Coronavirus infection (Trial version)Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Passengers' destinations from China: low risk of Novel Coronavirus (2019-nCoV) transmission into Africa and South AmericaEpidemiology and Infection\",\"A Simulation on Potential Secondary Spread of Novel Coronavirus in an Exported Country Using a Stochastic Epidemic SEIR ModelPreprints (www.preprints.org)\",\"Homologous recombination within the spike glycoprotein of the newly identified coronavirus may boost cross-species transmission from snake to humanJournal of medical virology\",\"A rapid advice guideline for the diagnosis and treatment of 2019 novel coronavirus (2019-nCoV) infected pneumonia (standard version)Military Medical Research\",\"Real-Time Estimation of the Risk of Death from Novel Coronavirus (COVID-19) Infection: Inference Using Exported CasesJournal of Clinical Medicine 2020, Vol. 9, Page 523\",\"Epidemiologic characteristics of early cases with 2019 novel coronavirus (2019-nCoV) disease in Republic of KoreaEpidemiology and Health\",\"The First Case of 2019 Novel Coronavirus Pneumonia Imported into Korea from Wuhan, China: Implication for Infection Prevention and Control MeasuresJournal of Korean medical science\",\"Mast cells contribute to coronavirus-induced inflammation: new anti-inflammatory strategyJournal of biological regulators and homeostatic agents\",\"Early Phylogenetic Estimate Of The Effective Reproduction Number Of 2019-nCoVmedRxiv\",\"Coronavirus Infections and Immune ResponsesJournal of medical virology\",\"OUP accepted manuscriptInternational Health\",\"Regulation of the ER Stress Response by the Ion Channel Activity of the Infectious Bronchitis Coronavirus Envelope Protein Modulates Virion Release, Apoptosis, Viral Fitness, and PathogenesisFrontiers in microbiology\",\"A Simple Prediction Model for the Development Trend of 2019-nCov Epidemics Based on Medical ObservationsarXiv preprint arXiv:2002.00426\",\"Interpretation of \\\"Guidelines for the Diagnosis and Treatment of Novel Coronavirus (2019-nCoV) Infection by the National Health Commission (Trial Version 5)\\\"Zhonghua yi xue za zhi\",\"The reproductive number of COVID-19 is higher compared to SARS coronavirusJournal of Travel Medicine\",\"Emerging novel Coronavirus (2019-nCoV) - Current scenario, evolutionary perspective based on genome analysis and recent developmentsVeterinary Quarterly\",\"The Rate of Underascertainment of Novel Coronavirus (2019-nCoV) Infection: Estimation Using Japanese Passengers Data on Evacuation FlightsJournal of clinical medicine\",\"The Extent of Transmission of Novel Coronavirus in Wuhan, China, 2020Journal of Clinical Medicine\",\"Initial CT findings and temporal changes in patients with the novel coronavirus pneumonia (2019-nCoV): a study of 63 patients in Wuhan, ChinaEuropean Radiology\",\"Full-genome evolutionary analysis of the novel corona virus (2019-nCoV) rejects the hypothesis of emergence as a result of a recent recombination eventInfection, Genetics and Evolution\",\"Novel coronavirus: From discovery to clinical diagnosticsInfection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"2019-nCoV (Wuhan virus), a novel Coronavirus: Human-to-human transmission, travel-related cases, and vaccine readinessJournal of Infection in Developing Countries\",\"Short-term Forecasts of the COVID-19 Epidemic in Guangdong and Zhejiang, China: February 13\\u201323, 2020Journal of Clinical Medicine\",\"Potential Maternal and Infant Outcomes from (Wuhan) Coronavirus 2019-nCoV Infecting Pregnant Women: Lessons from SARS, MERS, and Other Human Coronavirus InfectionsViruses\",\"Emergence of Novel Coronavirus 2019-nCoV: Need for Rapid Vaccine and Biologics DevelopmentPathogens\",\"2019 novel coronavirus of pneumonia in Wuhan, China: emerging attack and management strategiesClinical and Translational Medicine\",\"Genomic diversity of SARS-CoV-2 in Coronavirus Disease 2019 patientsClinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"Estimation of the Transmission Risk of the 2019-nCoV and Its Implication for Public Health InterventionsJournal of Clinical Medicine\",\"An updated estimation of the risk of transmission of the novel coronavirus (2019-nCov)Infectious Disease Modelling\",\"COVID-19, Australia: Epidemiology Report 2 (Reporting week ending 19:00 AEDT 8 February 2020)Communicable diseases intelligence (2018)\",\"Novel Coronavirus Outbreak in Wuhan, China, 2020: Intense Surveillance Is Vital for Preventing Sustained Transmission in New LocationsJournal of Clinical Medicine\",\"Potent binding of 2019 novel coronavirus spike protein by a SARS coronavirus-specific human monoclonal antibodyEmerging Microbes & Infections\",\"Consistent detection of 2019 novel coronavirus in salivaClinical Infectious Diseases\",\"Phase-adjusted estimation of the number of Coronavirus Disease 2019 cases in Wuhan, ChinaCell Discovery\",\"[Diagnosis and clinical management of 2019 novel coronavirus infection: an operational recommendation of Peking Union Medical College Hospital (V2.0)].Zhonghua nei ke za zhi\",\"Evolving status of the 2019 novel coronavirus Infection: proposal of conventional serologic assays for disease diagnosis and infection monitoring [Commentary/Review]Journal of medical virology\",\"In Vitro Antiviral Activity and Projection of Optimized Dosing Design of Hydroxychloroquine for the Treatment of Severe Acute Respiratory Syndrome Coronavirus 2 (SARS-CoV-2)Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"Measures for diagnosing and treating infections by a novel coronavirus responsible for a pneumonia outbreak originating in Wuhan, ChinaMicrobes and infection\",\"A familial cluster of infection associated with the 2019 novel coronavirus indicating potential person-to-person transmission during the incubation periodThe Journal of Infectious Diseases\",\"Potential Interventions for Novel Coronavirus in China: A Systemic ReviewJournal of Medical Virology\",\"Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replicationbioRxiv\",\"Molecular and serological investigation of 2019-nCoV infected patients: implication of multiple shedding routesEmerging Microbes & Infections\",\"Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 2020: A data-driven analysis in the early phase of the outbreakInternational journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Preliminary prediction of the basic reproduction number of the Wuhan novel coronavirus 2019-nCoVarXiv preprint arXiv:2001.10530\",\"The transmissibility of novel Coronavirus in the early stages of the 2019-20 outbreak in Wuhan: Exploring initial point-source exposure sizes and durations using scenario analysisWellcome Open Research\",\"The incubation period of 2019-nCoV infections among travellers from Wuhan, ChinamedRxiv\",\"The global spread of 2019-nCoV: a molecular evolutionary analysisPathogens and Global Health\",\"Pneumonia of Unknown Etiology in Wuhan, China: Potential for International Spread Via Commercial Air TravelJournal of travel medicine\",\"Risk Assessment of Novel Coronavirus COVID-19 Outbreaks Outside ChinaJournal of Clinical Medicine 2020, Vol. 9, Page 571\",\"Genomic characterization of the 2019 novel human-pathogenic coronavirus isolated from a patient with atypical pneumonia after visiting WuhanEmerging microbes & infections\",\"Pathogenicity and Transmissibility of 2019-nCoV-A Quick Overview and Comparison with Other Emerging VirusesMicrobes and infection\",\"Clinical progression of patients with COVID-19 in Shanghai, China.The Journal of infection\",\"Prediction of the SARS-CoV-2 (2019-nCoV) 3C-like protease (3CLpro) structure: virtual screening reveals velpatasvir, ledipasvir, and other drug repurposing candidatesF1000Research\",\"Emerging coronaviruses: genome structure, replication, and pathogenesisJournal of medical virology\",\"[Recommendations on extracorporeal life support for critically ill patients with novel coronavirus pneumonia].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Estimating the reproductive number and the outbreak size of Novel Coronavirus disease (COVID-19) using mathematical model in Republic of KoreaEpidemiology and Health\",\"Characteristics of and Public Health Responses to the Coronavirus Disease 2019 Outbreak in ChinaJournal of Clinical Medicine\",\"Genomic and protein structure modelling analysis depicts the origin and pathogenicity of 2019-nCoV, a new coronavirus which caused a pneumonia outbreak in Wuhan, ChinaF1000Research\",\"[Pharmacotherapeutics for the new coronavirus pneumonia].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Risk of 2019 novel coronavirus importations throughout China prior to the Wuhan quarantinemedRxiv\",\"The 2019 Novel Coronavirus (2019-nCoV): Novel Virus, Old ChallengesActa M\\u00e9dica Portuguesa\",\"The first two cases of 2019\\u2010nCoV in Italy: where they come from?Journal of Medical Virology\",\"Rigidity of the Outer Shell Predicted by a Protein Intrinsic Disorder Model Sheds Light on the COVID-19 (Wuhan-2019-nCoV) InfectivityBiomolecules\",\"Return of the Coronavirus: 2019-nCoVViruses\",\"Expert consensus for bronchoscopy during the epidemic of 2019 Novel Coronavirus infection (Trial version)Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Passengers' destinations from China: low risk of Novel Coronavirus (2019-nCoV) transmission into Africa and South AmericaEpidemiology and Infection\",\"In silico identification of vaccine targets for 2019-nCoVF1000Research\",\"A Simulation on Potential Secondary Spread of Novel Coronavirus in an Exported Country Using a Stochastic Epidemic SEIR ModelPreprints (www.preprints.org)\",\"Homologous recombination within the spike glycoprotein of the newly identified coronavirus may boost cross-species transmission from snake to humanJournal of medical virology\",\"A rapid advice guideline for the diagnosis and treatment of 2019 novel coronavirus (2019-nCoV) infected pneumonia (standard version)Military Medical Research\",\"Real-Time Estimation of the Risk of Death from Novel Coronavirus (COVID-19) Infection: Inference Using Exported CasesJournal of Clinical Medicine 2020, Vol. 9, Page 523\",\"Epidemiologic characteristics of early cases with 2019 novel coronavirus (2019-nCoV) disease in Republic of KoreaEpidemiology and Health\",\"The First Case of 2019 Novel Coronavirus Pneumonia Imported into Korea from Wuhan, China: Implication for Infection Prevention and Control MeasuresJournal of Korean medical science\",\"Mast cells contribute to coronavirus-induced inflammation: new anti-inflammatory strategyJournal of biological regulators and homeostatic agents\",\"Early Phylogenetic Estimate Of The Effective Reproduction Number Of 2019-nCoVmedRxiv\",\"Interrupting transmission of COVID-19: lessons from containment efforts in Singapore.Journal of travel medicine\",\"Coronavirus Infections and Immune ResponsesJournal of medical virology\",\"[Potential antiviral therapeutics for 2019 Novel Coronavirus].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"OUP accepted manuscriptInternational Health\",\"Regulation of the ER Stress Response by the Ion Channel Activity of the Infectious Bronchitis Coronavirus Envelope Protein Modulates Virion Release, Apoptosis, Viral Fitness, and PathogenesisFrontiers in microbiology\",\"A Simple Prediction Model for the Development Trend of 2019-nCov Epidemics Based on Medical ObservationsarXiv preprint arXiv:2002.00426\",\"Interpretation of \\\"Guidelines for the Diagnosis and Treatment of Novel Coronavirus (2019-nCoV) Infection by the National Health Commission (Trial Version 5)\\\"Zhonghua yi xue za zhi\",\"Clinical feature of COVID-19 in elderly patients: a comparison with young and middle-aged patients.The Journal of infection\",\"The reproductive number of COVID-19 is higher compared to SARS coronavirusJournal of Travel Medicine\",\"Emerging novel Coronavirus (2019-nCoV) - Current scenario, evolutionary perspective based on genome analysis and recent developmentsVeterinary Quarterly\",\"The Rate of Underascertainment of Novel Coronavirus (2019-nCoV) Infection: Estimation Using Japanese Passengers Data on Evacuation FlightsJournal of clinical medicine\",\"The Extent of Transmission of Novel Coronavirus in Wuhan, China, 2020Journal of Clinical Medicine\",\"Is lithium a potential treatment for the novel Wuhan (2019-nCoV) coronavirus? A scoping reviewF1000Research\",\"Initial CT findings and temporal changes in patients with the novel coronavirus pneumonia (2019-nCoV): a study of 63 patients in Wuhan, ChinaEuropean Radiology\",\"Full-genome evolutionary analysis of the novel corona virus (2019-nCoV) rejects the hypothesis of emergence as a result of a recent recombination eventInfection, Genetics and Evolution\",\"Novel coronavirus: From discovery to clinical diagnosticsInfection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"2019-nCoV (Wuhan virus), a novel Coronavirus: Human-to-human transmission, travel-related cases, and vaccine readinessJournal of Infection in Developing Countries\",\"Short-term Forecasts of the COVID-19 Epidemic in Guangdong and Zhejiang, China: February 13\\u201323, 2020Journal of Clinical Medicine\",\"Potential Maternal and Infant Outcomes from (Wuhan) Coronavirus 2019-nCoV Infecting Pregnant Women: Lessons from SARS, MERS, and Other Human Coronavirus InfectionsViruses\",\"Emergence of Novel Coronavirus 2019-nCoV: Need for Rapid Vaccine and Biologics DevelopmentPathogens\",\"2019 novel coronavirus of pneumonia in Wuhan, China: emerging attack and management strategiesClinical and Translational Medicine\",\"Genomic diversity of SARS-CoV-2 in Coronavirus Disease 2019 patientsClinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"[Epidemiological characteristics of 2019 novel coronavirus family clustering in Zhejiang Province].Zhonghua yu fang yi xue za zhi [Chinese journal of preventive medicine]\",\"Estimation of the Transmission Risk of the 2019-nCoV and Its Implication for Public Health InterventionsJournal of Clinical Medicine\",\"An updated estimation of the risk of transmission of the novel coronavirus (2019-nCov)Infectious Disease Modelling\",\"The effectiveness of quarantine and isolation determine the trend of the COVID-19 epidemics in the final phase of the current outbreak in China.International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"COVID-19, Australia: Epidemiology Report 2 (Reporting week ending 19:00 AEDT 8 February 2020)Communicable diseases intelligence (2018)\",\"Novel Coronavirus Outbreak in Wuhan, China, 2020: Intense Surveillance Is Vital for Preventing Sustained Transmission in New LocationsJournal of Clinical Medicine\",\"Potent binding of 2019 novel coronavirus spike protein by a SARS coronavirus-specific human monoclonal antibodyEmerging Microbes & Infections\",\"Consistent detection of 2019 novel coronavirus in salivaClinical Infectious Diseases\",\"Use of the informational spectrum methodology for rapid biological analysis of the novel coronavirus 2019-nCoV: prediction of potential receptor, natural reservoir, tropism and therapeutic/vaccine targetF1000Research\",\"Phase-adjusted estimation of the number of Coronavirus Disease 2019 cases in Wuhan, ChinaCell Discovery\",\"[Diagnosis and clinical management of 2019 novel coronavirus infection: an operational recommendation of Peking Union Medical College Hospital (V2.0)].Zhonghua nei ke za zhi\",\"Evolving status of the 2019 novel coronavirus Infection: proposal of conventional serologic assays for disease diagnosis and infection monitoring [Commentary/Review]Journal of medical virology\",\"In Vitro Antiviral Activity and Projection of Optimized Dosing Design of Hydroxychloroquine for the Treatment of Severe Acute Respiratory Syndrome Coronavirus 2 (SARS-CoV-2)Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"Clinical characteristics of severe acute respiratory syndrome coronavirus 2 reactivation.The Journal of infection\",\"Measures for diagnosing and treating infections by a novel coronavirus responsible for a pneumonia outbreak originating in Wuhan, ChinaMicrobes and infection\",\"A familial cluster of infection associated with the 2019 novel coronavirus indicating potential person-to-person transmission during the incubation periodThe Journal of Infectious Diseases\",\"Potential Interventions for Novel Coronavirus in China: A Systemic ReviewJournal of Medical Virology\",\"Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replicationbioRxiv\",\"Molecular and serological investigation of 2019-nCoV infected patients: implication of multiple shedding routesEmerging Microbes & Infections\",\"Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 2020: A data-driven analysis in the early phase of the outbreakInternational journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Preliminary prediction of the basic reproduction number of the Wuhan novel coronavirus 2019-nCoVarXiv preprint arXiv:2001.10530\",\"Short-term moderate-dose corticosteroid plus immunoglobulin effectively reverses COVID-19 patients who have failed low-dose therapyPreprints (www.preprints.org)\",\"Preliminary estimation of the novel coronavirus disease (COVID-19) cases in Iran: A modelling analysis based on overseas cases and air travel data.International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"The transmissibility of novel Coronavirus in the early stages of the 2019-20 outbreak in Wuhan: Exploring initial point-source exposure sizes and durations using scenario analysisWellcome Open Research\",\"The incubation period of 2019-nCoV infections among travellers from Wuhan, ChinamedRxiv\",\"The global spread of 2019-nCoV: a molecular evolutionary analysisPathogens and Global Health\",\"Application of the ARIMA model on the COVID-2019 epidemic datasetData in Brief\",\"Pneumonia of Unknown Etiology in Wuhan, China: Potential for International Spread Via Commercial Air TravelJournal of travel medicine\",\"Risk Assessment of Novel Coronavirus COVID-19 Outbreaks Outside ChinaJournal of Clinical Medicine 2020, Vol. 9, Page 571\",\"Genomic characterization of the 2019 novel human-pathogenic coronavirus isolated from a patient with atypical pneumonia after visiting WuhanEmerging microbes & infections\",\"Pathogenicity and Transmissibility of 2019-nCoV-A Quick Overview and Comparison with Other Emerging VirusesMicrobes and infection\",\"Clinical progression of patients with COVID-19 in Shanghai, China.The Journal of infection\",\"Prediction of the SARS-CoV-2 (2019-nCoV) 3C-like protease (3CLpro) structure: virtual screening reveals velpatasvir, ledipasvir, and other drug repurposing candidatesF1000Research\",\"Emerging coronaviruses: genome structure, replication, and pathogenesisJournal of medical virology\",\"[Recommendations on extracorporeal life support for critically ill patients with novel coronavirus pneumonia].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Estimating the reproductive number and the outbreak size of Novel Coronavirus disease (COVID-19) using mathematical model in Republic of KoreaEpidemiology and Health\",\"Characteristics of and Public Health Responses to the Coronavirus Disease 2019 Outbreak in ChinaJournal of Clinical Medicine\",\"Genomic and protein structure modelling analysis depicts the origin and pathogenicity of 2019-nCoV, a new coronavirus which caused a pneumonia outbreak in Wuhan, ChinaF1000Research\",\"[Pharmacotherapeutics for the new coronavirus pneumonia].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Risk of 2019 novel coronavirus importations throughout China prior to the Wuhan quarantinemedRxiv\",\"The 2019 Novel Coronavirus (2019-nCoV): Novel Virus, Old ChallengesActa M\\u00e9dica Portuguesa\",\"The first two cases of 2019\\u2010nCoV in Italy: where they come from?Journal of Medical Virology\",\"Rigidity of the Outer Shell Predicted by a Protein Intrinsic Disorder Model Sheds Light on the COVID-19 (Wuhan-2019-nCoV) InfectivityBiomolecules\",\"Return of the Coronavirus: 2019-nCoVViruses\",\"Expert consensus for bronchoscopy during the epidemic of 2019 Novel Coronavirus infection (Trial version)Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"Passengers' destinations from China: low risk of Novel Coronavirus (2019-nCoV) transmission into Africa and South AmericaEpidemiology and Infection\",\"In silico identification of vaccine targets for 2019-nCoVF1000Research\",\"A Simulation on Potential Secondary Spread of Novel Coronavirus in an Exported Country Using a Stochastic Epidemic SEIR ModelPreprints (www.preprints.org)\",\"Homologous recombination within the spike glycoprotein of the newly identified coronavirus may boost cross-species transmission from snake to humanJournal of medical virology\",\"A rapid advice guideline for the diagnosis and treatment of 2019 novel coronavirus (2019-nCoV) infected pneumonia (standard version)Military Medical Research\",\"Real-Time Estimation of the Risk of Death from Novel Coronavirus (COVID-19) Infection: Inference Using Exported CasesJournal of Clinical Medicine 2020, Vol. 9, Page 523\",\"Epidemiologic characteristics of early cases with 2019 novel coronavirus (2019-nCoV) disease in Republic of KoreaEpidemiology and Health\",\"The First Case of 2019 Novel Coronavirus Pneumonia Imported into Korea from Wuhan, China: Implication for Infection Prevention and Control MeasuresJournal of Korean medical science\",\"Mast cells contribute to coronavirus-induced inflammation: new anti-inflammatory strategyJournal of biological regulators and homeostatic agents\",\"Early Phylogenetic Estimate Of The Effective Reproduction Number Of 2019-nCoVmedRxiv\",\"The positive impact of lockdown in Wuhan on containing the COVID-19 outbreak in ChinaJournal of Travel Medicine\",\"Interrupting transmission of COVID-19: lessons from containment efforts in Singapore.Journal of travel medicine\",\"Coronavirus Infections and Immune ResponsesJournal of medical virology\",\"[Potential antiviral therapeutics for 2019 Novel Coronavirus].Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\",\"OUP accepted manuscriptInternational Health\",\"Regulation of the ER Stress Response by the Ion Channel Activity of the Infectious Bronchitis Coronavirus Envelope Protein Modulates Virion Release, Apoptosis, Viral Fitness, and PathogenesisFrontiers in microbiology\",\"A Simple Prediction Model for the Development Trend of 2019-nCov Epidemics Based on Medical ObservationsarXiv preprint arXiv:2002.00426\",\"Interpretation of \\\"Guidelines for the Diagnosis and Treatment of Novel Coronavirus (2019-nCoV) Infection by the National Health Commission (Trial Version 5)\\\"Zhonghua yi xue za zhi\",\"Clinical feature of COVID-19 in elderly patients: a comparison with young and middle-aged patients.The Journal of infection\",\"The reproductive number of COVID-19 is higher compared to SARS coronavirusJournal of Travel Medicine\",\"Emerging novel Coronavirus (2019-nCoV) - Current scenario, evolutionary perspective based on genome analysis and recent developmentsVeterinary Quarterly\",\"Transmission potential of the novel coronavirus (COVID-19) onboard the diamond Princess Cruises Ship, 2020Infectious Disease Modelling\",\"The Rate of Underascertainment of Novel Coronavirus (2019-nCoV) Infection: Estimation Using Japanese Passengers Data on Evacuation FlightsJournal of clinical medicine\",\"The Extent of Transmission of Novel Coronavirus in Wuhan, China, 2020Journal of Clinical Medicine\",\"Is lithium a potential treatment for the novel Wuhan (2019-nCoV) coronavirus? A scoping reviewF1000Research\",\"Initial CT findings and temporal changes in patients with the novel coronavirus pneumonia (2019-nCoV): a study of 63 patients in Wuhan, ChinaEuropean Radiology\",\"Full-genome evolutionary analysis of the novel corona virus (2019-nCoV) rejects the hypothesis of emergence as a result of a recent recombination eventInfection, Genetics and Evolution\",\"Novel coronavirus: From discovery to clinical diagnosticsInfection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases\",\"2019-nCoV (Wuhan virus), a novel Coronavirus: Human-to-human transmission, travel-related cases, and vaccine readinessJournal of Infection in Developing Countries\",\"COVID-19 outbreak on the Diamond Princess cruise ship: estimating the epidemic potential and effectiveness of public health countermeasuresJournal of Travel Medicine\",\"Short-term Forecasts of the COVID-19 Epidemic in Guangdong and Zhejiang, China: February 13\\u201323, 2020Journal of Clinical Medicine\",\"Potential Maternal and Infant Outcomes from (Wuhan) Coronavirus 2019-nCoV Infecting Pregnant Women: Lessons from SARS, MERS, and Other Human Coronavirus InfectionsViruses\",\"Emergence of Novel Coronavirus 2019-nCoV: Need for Rapid Vaccine and Biologics DevelopmentPathogens\",\"2019 novel coronavirus of pneumonia in Wuhan, China: emerging attack and management strategiesClinical and Translational Medicine\",\"Genomic diversity of SARS-CoV-2 in Coronavirus Disease 2019 patientsClinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"[Epidemiological characteristics of 2019 novel coronavirus family clustering in Zhejiang Province].Zhonghua yu fang yi xue za zhi [Chinese journal of preventive medicine]\",\"Estimation of the Transmission Risk of the 2019-nCoV and Its Implication for Public Health InterventionsJournal of Clinical Medicine\",\"An updated estimation of the risk of transmission of the novel coronavirus (2019-nCov)Infectious Disease Modelling\",\"The effectiveness of quarantine and isolation determine the trend of the COVID-19 epidemics in the final phase of the current outbreak in China.International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"COVID-19, Australia: Epidemiology Report 2 (Reporting week ending 19:00 AEDT 8 February 2020)Communicable diseases intelligence (2018)\",\"Novel Coronavirus Outbreak in Wuhan, China, 2020: Intense Surveillance Is Vital for Preventing Sustained Transmission in New LocationsJournal of Clinical Medicine\",\"Potent binding of 2019 novel coronavirus spike protein by a SARS coronavirus-specific human monoclonal antibodyEmerging Microbes & Infections\",\"Consistent detection of 2019 novel coronavirus in salivaClinical Infectious Diseases\",\"Use of the informational spectrum methodology for rapid biological analysis of the novel coronavirus 2019-nCoV: prediction of potential receptor, natural reservoir, tropism and therapeutic/vaccine targetF1000Research\",\"Phase-adjusted estimation of the number of Coronavirus Disease 2019 cases in Wuhan, ChinaCell Discovery\",\"[Diagnosis and clinical management of 2019 novel coronavirus infection: an operational recommendation of Peking Union Medical College Hospital (V2.0)].Zhonghua nei ke za zhi\",\"Evolving status of the 2019 novel coronavirus Infection: proposal of conventional serologic assays for disease diagnosis and infection monitoring [Commentary/Review]Journal of medical virology\",\"In Vitro Antiviral Activity and Projection of Optimized Dosing Design of Hydroxychloroquine for the Treatment of Severe Acute Respiratory Syndrome Coronavirus 2 (SARS-CoV-2)Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\",\"Clinical characteristics of severe acute respiratory syndrome coronavirus 2 reactivation.The Journal of infection\",\"Measures for diagnosing and treating infections by a novel coronavirus responsible for a pneumonia outbreak originating in Wuhan, ChinaMicrobes and infection\",\"A familial cluster of infection associated with the 2019 novel coronavirus indicating potential person-to-person transmission during the incubation periodThe Journal of Infectious Diseases\",\"Potential Interventions for Novel Coronavirus in China: A Systemic ReviewJournal of Medical Virology\",\"Alpha-ketoamides as broad-spectrum inhibitors of coronavirus and enterovirus replicationbioRxiv\",\"Molecular and serological investigation of 2019-nCoV infected patients: implication of multiple shedding routesEmerging Microbes & Infections\",\"Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 2020: A data-driven analysis in the early phase of the outbreakInternational journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\",\"Preliminary prediction of the basic reproduction number of the Wuhan novel coronavirus 2019-nCoVarXiv preprint arXiv:2001.10530\",\"Short-term moderate-dose corticosteroid plus immunoglobulin effectively reverses COVID-19 patients who have failed low-dose therapyPreprints (www.preprints.org)\",\"Preliminary estimation of the novel coronavirus disease (COVID-19) cases in Iran: A modelling analysis based on overseas cases and air travel data.International journal of infectious diseases : IJID : official publication of the International Society for Infectious Diseases\"],\"x\":{\"__ndarray__\":\"U6sowViFUUHSelw/S8+ZwMLcQ8HH43JBUUV3v7h/AUFr/0nAQ7xmQbXkAMHC9ZLAyxeKP0bLsD8wZdPAyFUzwTjRN8FFP4HBI4V5wbo3PUAJlB1Br1zkwBBlGkGXVPJArqRTQA5zQMAXUJ0/GulxQOUZoEDZZ4nBzg0zwLLT8r9hZibBkGFswZZIS0GBZWE/9QCRwNk+ScFtOHhBilmgv+FRCUHilmHAT79vQYuTBMGarKPA747rP3G72T9/ocrAab05wZdaNsEhFHjBDuh7wcurQUDcFxJBHjYNwQT/0sCIPRxBI98CQVCVhsDV6VVAErdTwLbahkAo+jRAlnyVwEAs9EDarYJBotxePzBmZkAzPKBARjwCQUAHicEs8GHA86qzv97fBkHA3jLBWuxkwX0nUEF7gbLADQyKPyB2ksCYsEDBngh0QcmRgj49rrtAnON1v6yjAkEQdQXA+oJwwJdbakFwrgXBUtidwFE6sj6aqLs/NDLVwNsnOMFh5DTBENl9wdmVgsFshmBAlZsdQTV9CcFVS+TAbo0lQUlA/ECIeIbADQZ5QLSbN8BN2gtAlZ+DQMFAF0DXkEZAiXCdwJV78EDRcTbBe4iDQWWILz+BmodAwV+XQAZilkD5GglBbQOPwVx4BMF/pEPAz8TFv/dDQ0HeQ0NBvDvVQEsMM8Hj84HB+GOHwOvrP8GoOO6/Q4aOwNPtpMBftYvBxoFFwcS1AsFJJa7AAEB9wA2DiUBJNTTBYAgiPj2uu0CX4pA+dk4pQK2gcMBQIiJBPYYLQdV4wsDnzo8/nRH+QCjkAcAGYpZAxhoywBpBFEEyZPdAd0PcP0l8gMF/bx+/3pVFQfAleL+d2E9A/mp8QI//uUCKnG9BdgF3QMBAF0De3wZBFbrSwJBQpz9FIJDAgYsBQXP9LcHY/wpBNU8LQMO/k0C/VAHBBq0RwetTZMGnxWNBX/uEQcYQS8De3wZBnUkuwR36bMEQwEZBFE21wDAzuT8LFpnA6+8+wZEmeUGhboI+Pa67QNpEsL8k0wxBKOQBwFhjQMCSFmxBhv+5QDO5pMBGXQnBf7GjwM19sj6PO2Y/FYzYwNzWLsH2lCjBTp9AwVT4eMFMvHvB2aM2QANmFEGQPQnBI2PcwG9gIUEgjQJBAEB9wG8FTECtdyrANU8LQH9vH7/oDo1AvDvVQMRAF0BA/9o943tDQMQZC0HTxI3AdHv/QJZWNcH19YFBGD0cwMyTKT+JB4FAJ2MjQNcLmEAGYpZAYLMFQXqJjcF3wgLBFz5EwDpwA8De3wZB8sEvwQ3WZ8FITk5Be4GywK8GyT9BkYTAvcC8v1nYrEC2qD3BUxxvQYoJgT/PkYI+I685Pj/IOD89rrtAREepv68/B0Eo5AHAwOJfwLvfbkGP/7lA87NwwMLHpMCD3gvBSxOTwMEIIj5QcZA/ELnDwA7XNsFwlSjBnsTRwG9iO8F9OKq/C1OAwR+JgsFpbFVANmsZQS3J7MD8qw3BccDOwDrkHEG0mO5AJmkWwfMMb8DdR2VABDdYwDVPC0BIXh+/6st7QLw71UDEQBdAQP/aPQ8+ZUAstitAxBkLQR5eXT3wOJ7Antr+QHotM8Fg0IVBLuvnvycEHMAef3U/K9uDQGpnI0CH3ArA5KmnQAZilkBtSgZBy9uKwR6EBMFQ/1jAtkKvv9jBX8DuWTzA+i0DQfuXJ8GoO2nBXvgNwJo0SUEUTbXAxsKyP2OeicC9wLy/UNisQF2wSMGkAntBbQmBP8uRgj7Fizg+HJ44Pz2uu0CwE3i/zKgEQRB1BcCbO0zAhU1lQU8BukDcs3DA0+2kwPU8CsHQtJzAhAgiPgPhXz95RcvArXkxwQ2VKMGUC/I9ssTRwEAlPcH3S6q/k357wYNugMGL9GNARmUZQVzK7MBrbhHBxlHcwGDvqL/u6CVBcsfzQCRpFsFjc3fAtjZqQJcUMMBMGwpAUxvSwEheH7/hHIxAvDvVQMFAF0BA/9o9Kj1lQFSAO0AtAQtBz99cPQOLlsDdfPBA3FkzwTGIhkH77ee/MTwcwNOqlD/gMG5AkGQjQBbKCsAfDqdABmKWQP2/CUHdOY7BL3UEwYOEWMAKks6/Rr1fwGRXPMA=\",\"dtype\":\"float32\",\"shape\":[398]},\"x_backup\":{\"__ndarray__\":\"U6sowViFUUHSelw/S8+ZwMLcQ8HH43JBUUV3v7h/AUFr/0nAQ7xmQbXkAMHC9ZLAyxeKP0bLsD8wZdPAyFUzwTjRN8FFP4HBI4V5wbo3PUAJlB1Br1zkwBBlGkGXVPJArqRTQA5zQMAXUJ0/GulxQOUZoEDZZ4nBzg0zwLLT8r9hZibBkGFswZZIS0GBZWE/9QCRwNk+ScFtOHhBilmgv+FRCUHilmHAT79vQYuTBMGarKPA747rP3G72T9/ocrAab05wZdaNsEhFHjBDuh7wcurQUDcFxJBHjYNwQT/0sCIPRxBI98CQVCVhsDV6VVAErdTwLbahkAo+jRAlnyVwEAs9EDarYJBotxePzBmZkAzPKBARjwCQUAHicEs8GHA86qzv97fBkHA3jLBWuxkwX0nUEF7gbLADQyKPyB2ksCYsEDBngh0QcmRgj49rrtAnON1v6yjAkEQdQXA+oJwwJdbakFwrgXBUtidwFE6sj6aqLs/NDLVwNsnOMFh5DTBENl9wdmVgsFshmBAlZsdQTV9CcFVS+TAbo0lQUlA/ECIeIbADQZ5QLSbN8BN2gtAlZ+DQMFAF0DXkEZAiXCdwJV78EDRcTbBe4iDQWWILz+BmodAwV+XQAZilkD5GglBbQOPwVx4BMF/pEPAz8TFv/dDQ0HeQ0NBvDvVQEsMM8Hj84HB+GOHwOvrP8GoOO6/Q4aOwNPtpMBftYvBxoFFwcS1AsFJJa7AAEB9wA2DiUBJNTTBYAgiPj2uu0CX4pA+dk4pQK2gcMBQIiJBPYYLQdV4wsDnzo8/nRH+QCjkAcAGYpZAxhoywBpBFEEyZPdAd0PcP0l8gMF/bx+/3pVFQfAleL+d2E9A/mp8QI//uUCKnG9BdgF3QMBAF0De3wZBFbrSwJBQpz9FIJDAgYsBQXP9LcHY/wpBNU8LQMO/k0C/VAHBBq0RwetTZMGnxWNBX/uEQcYQS8De3wZBnUkuwR36bMEQwEZBFE21wDAzuT8LFpnA6+8+wZEmeUGhboI+Pa67QNpEsL8k0wxBKOQBwFhjQMCSFmxBhv+5QDO5pMBGXQnBf7GjwM19sj6PO2Y/FYzYwNzWLsH2lCjBTp9AwVT4eMFMvHvB2aM2QANmFEGQPQnBI2PcwG9gIUEgjQJBAEB9wG8FTECtdyrANU8LQH9vH7/oDo1AvDvVQMRAF0BA/9o943tDQMQZC0HTxI3AdHv/QJZWNcH19YFBGD0cwMyTKT+JB4FAJ2MjQNcLmEAGYpZAYLMFQXqJjcF3wgLBFz5EwDpwA8De3wZB8sEvwQ3WZ8FITk5Be4GywK8GyT9BkYTAvcC8v1nYrEC2qD3BUxxvQYoJgT/PkYI+I685Pj/IOD89rrtAREepv68/B0Eo5AHAwOJfwLvfbkGP/7lA87NwwMLHpMCD3gvBSxOTwMEIIj5QcZA/ELnDwA7XNsFwlSjBnsTRwG9iO8F9OKq/C1OAwR+JgsFpbFVANmsZQS3J7MD8qw3BccDOwDrkHEG0mO5AJmkWwfMMb8DdR2VABDdYwDVPC0BIXh+/6st7QLw71UDEQBdAQP/aPQ8+ZUAstitAxBkLQR5eXT3wOJ7Antr+QHotM8Fg0IVBLuvnvycEHMAef3U/K9uDQGpnI0CH3ArA5KmnQAZilkBtSgZBy9uKwR6EBMFQ/1jAtkKvv9jBX8DuWTzA+i0DQfuXJ8GoO2nBXvgNwJo0SUEUTbXAxsKyP2OeicC9wLy/UNisQF2wSMGkAntBbQmBP8uRgj7Fizg+HJ44Pz2uu0CwE3i/zKgEQRB1BcCbO0zAhU1lQU8BukDcs3DA0+2kwPU8CsHQtJzAhAgiPgPhXz95RcvArXkxwQ2VKMGUC/I9ssTRwEAlPcH3S6q/k357wYNugMGL9GNARmUZQVzK7MBrbhHBxlHcwGDvqL/u6CVBcsfzQCRpFsFjc3fAtjZqQJcUMMBMGwpAUxvSwEheH7/hHIxAvDvVQMFAF0BA/9o9Kj1lQFSAO0AtAQtBz99cPQOLlsDdfPBA3FkzwTGIhkH77ee/MTwcwNOqlD/gMG5AkGQjQBbKCsAfDqdABmKWQP2/CUHdOY7BL3UEwYOEWMAKks6/Rr1fwGRXPMA=\",\"dtype\":\"float32\",\"shape\":[398]},\"y\":{\"__ndarray__\":\"ZX8zQUvEFcA86gnBUGeUwF3eSMDWVAjBsANeP+tOpj8TY2XBsO3aQHXdi8CEKbO/w+GrQHdKqkDi8ypAjf5Gwc+88MA/kBQ/08tKwQb6dkG1ng/Begf7wKV8M0GOpYBBjGwzwUXkHMFUqKHA8ng5wKQfUT/F3fjA/I+iwc1Y/kCUzC1Bw3WWQAEB8r/xdQXBFLejwNkMaMDFnQfB4RCqPyVAfT+F9W/BSUviQLvShMDmbaS/Vbu3QLKlrUCRxyxA3RhAwbNRAcEvNmc+7e5FwelEgEFjqQzBnH2Wwf9198BQ7S5BQoF6QQkRkUASqjzB+ikfwS8CtcDyjipByjN7QULCokDSfh1AGxKRwPTtLcCQ5r8/yjyKwC1a7sDd7qPBWfv8QPYAX8HMQCtBMEmhQC1sAcApmzNB3cwMwaocjsASQnTAuK/4wKhs67+aOENBV91eP/A6yD/D/D/AeZBswbjM8UBNU5vAqxNFv2hiTkHHAsdAtDQIQPghRcGhnPnARlMzPz02SMFU0XdBqwAKwdQgmsHPswLBqhYwQcBegUE0SpFAML05wTKLJ8G3y1e+Q2nEwLqaJEBeriRBxhV9QfOBlUDJPo8/++E8QI/6oMA4YiDAI/iyP7dcm0AAr33AVFDwwNNyQb8w+KbBMhvsQCzzMUBt8zFAxMu8v+D/MEFV6Hw+e6GgwBR5AMF+Hu1AByuBQft780CYT/7AALx1wFWGYr8CDzFBy3uFQPMcNMAqVJ4/2FlOQZo4Q0HSXO+/Qy0oQeFLZ8E2yzhBYnnFPxJsHEDLcwLBmRt3QZ4TRMC3XJtAry6lwUg8CMF8w5NAxC3BQGnURMHOG4RBV/8WwNLWqT8WOnVBw3W4wGNtB0EJi/7ALMU0wbqaJED2AF/BxocEwROCmMD9ro6/xaKCwJsKP8FKdvhA2DtivuxfjT+iKpXAUOaZwU0/mEDYgONAY4kZQIKiJ8H2AF/BT7Q0Qdsdn0BncwLAeTIwQWleA8GwY5/Ay8JPwOPu+8DZY+u/mjhDQTE3jD9m0Z4/nhNEwCo8a8F5adpAYm0HQclt80CRBYbAPnt9v7lhTkF2H75Ao1kbQHhRRMFufRNB7qv2wNdgCz/4qk/Buh18QaKIEcGKypfBUNQFwURULUGpqX9By3uFQBo0OMHSRyPB2Dtivs4bhEHIeLnAxMu8v8SaJECAUDjBvzIpQf99+ECw3X1BBPmZQAamfz+/4C5AgO4nQe1/l8C8dxPANnuCwdbYWz+3XJtAGHd4wJN058BigmO/l/WgwVen9kD2AF/BtRQqQYvFkkCxmibAKZszQQVwCMHILprA85OUQJ+6DcHaxWLAaaYEwUIDCEGlbOu/zvY+QPGZZcGaOENBdNxzP4uJ2T+eE0TAmDFkwf207EBibQdBbAruwNxU80CWyY7AzthYv7FZTkHgC8VAlDAMQDzgO8GpgBNB7L09wekUA8HN4ejAYNtWvGaHTcG3hYBBEMISwbNtZ8FXL5vB8hoAwVXpN0HWknxBN4ucQOjwjUA77zHBobQjwdk7Yr4cIoRBTIK/wMTLvL/EmiRAgFA4wSs36EDnqCNB/334QPu+JUElxYBBAXKeQGKAmz+pSDlAmnmxwNzxJ0GCQaXAMBY+wHd6gsEDTT+/Bc2CP7dcm0DZmo3AH4rmwJ8lPb80iKHBddj0QB60kD9lzlNB9x1hwQvHK0HI9aRAbLwDQI4oJ8B5MjBBqvkLwcM8kcDzk5RAn7oNwc79U8Bw/gLBUAMIQals679D2z5Aq35lwZo4Q0EWz6k/RWmCP8P8P8BbsW/BrKrtQG1tB0FLCu7A+3vzQFBXmMCwHr+/2llOQWdrtEBikwJAHHk7wW59E0F6kJM/6Lw9wWTT78ARAOnAAkfwvMXgUMEoS31BU/AGwRdtZ8F7jZfBbCv1wL5UKEAEWjVB/AB4QQ2LnEBU/5RAyG09wRHEHsGVOVK+77WsQBwihEGGA8HAxMu8v7qaJECAUDjBsC/oQMJ0IkEpfvhALL0lQVMfgkGr7p9A+2CBP+2uJ0DBe7HAfu4nQa/gkMCRERzAznqCwVrNPr8w76s/t1ybQA6JiMAjkfrAUZs7v2NppsGGzQBBZrKQP7LFU0E=\",\"dtype\":\"float32\",\"shape\":[398]},\"y_backup\":{\"__ndarray__\":\"ZX8zQUvEFcA86gnBUGeUwF3eSMDWVAjBsANeP+tOpj8TY2XBsO3aQHXdi8CEKbO/w+GrQHdKqkDi8ypAjf5Gwc+88MA/kBQ/08tKwQb6dkG1ng/Begf7wKV8M0GOpYBBjGwzwUXkHMFUqKHA8ng5wKQfUT/F3fjA/I+iwc1Y/kCUzC1Bw3WWQAEB8r/xdQXBFLejwNkMaMDFnQfB4RCqPyVAfT+F9W/BSUviQLvShMDmbaS/Vbu3QLKlrUCRxyxA3RhAwbNRAcEvNmc+7e5FwelEgEFjqQzBnH2Wwf9198BQ7S5BQoF6QQkRkUASqjzB+ikfwS8CtcDyjipByjN7QULCokDSfh1AGxKRwPTtLcCQ5r8/yjyKwC1a7sDd7qPBWfv8QPYAX8HMQCtBMEmhQC1sAcApmzNB3cwMwaocjsASQnTAuK/4wKhs67+aOENBV91eP/A6yD/D/D/AeZBswbjM8UBNU5vAqxNFv2hiTkHHAsdAtDQIQPghRcGhnPnARlMzPz02SMFU0XdBqwAKwdQgmsHPswLBqhYwQcBegUE0SpFAML05wTKLJ8G3y1e+Q2nEwLqaJEBeriRBxhV9QfOBlUDJPo8/++E8QI/6oMA4YiDAI/iyP7dcm0AAr33AVFDwwNNyQb8w+KbBMhvsQCzzMUBt8zFAxMu8v+D/MEFV6Hw+e6GgwBR5AMF+Hu1AByuBQft780CYT/7AALx1wFWGYr8CDzFBy3uFQPMcNMAqVJ4/2FlOQZo4Q0HSXO+/Qy0oQeFLZ8E2yzhBYnnFPxJsHEDLcwLBmRt3QZ4TRMC3XJtAry6lwUg8CMF8w5NAxC3BQGnURMHOG4RBV/8WwNLWqT8WOnVBw3W4wGNtB0EJi/7ALMU0wbqaJED2AF/BxocEwROCmMD9ro6/xaKCwJsKP8FKdvhA2DtivuxfjT+iKpXAUOaZwU0/mEDYgONAY4kZQIKiJ8H2AF/BT7Q0Qdsdn0BncwLAeTIwQWleA8GwY5/Ay8JPwOPu+8DZY+u/mjhDQTE3jD9m0Z4/nhNEwCo8a8F5adpAYm0HQclt80CRBYbAPnt9v7lhTkF2H75Ao1kbQHhRRMFufRNB7qv2wNdgCz/4qk/Buh18QaKIEcGKypfBUNQFwURULUGpqX9By3uFQBo0OMHSRyPB2Dtivs4bhEHIeLnAxMu8v8SaJECAUDjBvzIpQf99+ECw3X1BBPmZQAamfz+/4C5AgO4nQe1/l8C8dxPANnuCwdbYWz+3XJtAGHd4wJN058BigmO/l/WgwVen9kD2AF/BtRQqQYvFkkCxmibAKZszQQVwCMHILprA85OUQJ+6DcHaxWLAaaYEwUIDCEGlbOu/zvY+QPGZZcGaOENBdNxzP4uJ2T+eE0TAmDFkwf207EBibQdBbAruwNxU80CWyY7AzthYv7FZTkHgC8VAlDAMQDzgO8GpgBNB7L09wekUA8HN4ejAYNtWvGaHTcG3hYBBEMISwbNtZ8FXL5vB8hoAwVXpN0HWknxBN4ucQOjwjUA77zHBobQjwdk7Yr4cIoRBTIK/wMTLvL/EmiRAgFA4wSs36EDnqCNB/334QPu+JUElxYBBAXKeQGKAmz+pSDlAmnmxwNzxJ0GCQaXAMBY+wHd6gsEDTT+/Bc2CP7dcm0DZmo3AH4rmwJ8lPb80iKHBddj0QB60kD9lzlNB9x1hwQvHK0HI9aRAbLwDQI4oJ8B5MjBBqvkLwcM8kcDzk5RAn7oNwc79U8Bw/gLBUAMIQals679D2z5Aq35lwZo4Q0EWz6k/RWmCP8P8P8BbsW/BrKrtQG1tB0FLCu7A+3vzQFBXmMCwHr+/2llOQWdrtEBikwJAHHk7wW59E0F6kJM/6Lw9wWTT78ARAOnAAkfwvMXgUMEoS31BU/AGwRdtZ8F7jZfBbCv1wL5UKEAEWjVB/AB4QQ2LnEBU/5RAyG09wRHEHsGVOVK+77WsQBwihEGGA8HAxMu8v7qaJECAUDjBsC/oQMJ0IkEpfvhALL0lQVMfgkGr7p9A+2CBP+2uJ0DBe7HAfu4nQa/gkMCRERzAznqCwVrNPr8w76s/t1ybQA6JiMAjkfrAUZs7v2NppsGGzQBBZrKQP7LFU0E=\",\"dtype\":\"float32\",\"shape\":[398]}},\"selected\":{\"id\":\"1243\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"1244\",\"type\":\"UnionRenderers\"}},\"id\":\"1177\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"1187\",\"type\":\"LinearScale\"},{\"attributes\":{},\"id\":\"1197\",\"type\":\"BasicTicker\"},{\"attributes\":{\"fill_color\":{\"field\":\"desc\",\"transform\":{\"id\":\"1179\",\"type\":\"LinearColorMapper\"}},\"line_alpha\":{\"value\":0.3},\"line_color\":{\"value\":\"#1f77b4\"},\"size\":{\"units\":\"screen\",\"value\":5},\"x\":{\"field\":\"x\"},\"y\":{\"field\":\"y\"}},\"id\":\"1216\",\"type\":\"Scatter\"},{\"attributes\":{\"children\":[{\"id\":\"1180\",\"subtype\":\"Figure\",\"type\":\"Plot\"}]},\"id\":\"1864\",\"type\":\"Row\"},{\"attributes\":{},\"id\":\"1243\",\"type\":\"Selection\"},{\"attributes\":{\"below\":[{\"id\":\"1191\",\"type\":\"LinearAxis\"}],\"center\":[{\"id\":\"1195\",\"type\":\"Grid\"},{\"id\":\"1200\",\"type\":\"Grid\"},{\"id\":\"1226\",\"type\":\"Legend\"}],\"left\":[{\"id\":\"1196\",\"type\":\"LinearAxis\"}],\"plot_height\":850,\"plot_width\":1200,\"renderers\":[{\"id\":\"1218\",\"type\":\"GlyphRenderer\"}],\"title\":{\"id\":\"1181\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"1207\",\"type\":\"Toolbar\"},\"toolbar_location\":\"above\",\"x_range\":{\"id\":\"1183\",\"type\":\"DataRange1d\"},\"x_scale\":{\"id\":\"1187\",\"type\":\"LinearScale\"},\"y_range\":{\"id\":\"1185\",\"type\":\"DataRange1d\"},\"y_scale\":{\"id\":\"1189\",\"type\":\"LinearScale\"}},\"id\":\"1180\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"data_source\":{\"id\":\"1177\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"1216\",\"type\":\"Scatter\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"1217\",\"type\":\"Scatter\"},\"selection_glyph\":null,\"view\":{\"id\":\"1219\",\"type\":\"CDSView\"}},\"id\":\"1218\",\"type\":\"GlyphRenderer\"},{\"attributes\":{},\"id\":\"1201\",\"type\":\"PanTool\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"size\":{\"units\":\"screen\",\"value\":5},\"x\":{\"field\":\"x\"},\"y\":{\"field\":\"y\"}},\"id\":\"1217\",\"type\":\"Scatter\"}],\"root_ids\":[\"1865\"]},\"title\":\"Bokeh Application\",\"version\":\"1.3.4\"}};\n  var render_items = [{\"docid\":\"842baa40-3731-4afa-bd8a-a9dac2d13a25\",\"roots\":{\"1865\":\"dbcfc130-9bb4-45f6-b42f-0183c785354e\"}}];\n  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n\n  }\n  if (root.Bokeh !== undefined) {\n    embed_document(root);\n  } else {\n    var attempts = 0;\n    var timer = setInterval(function(root) {\n      if (root.Bokeh !== undefined) {\n        embed_document(root);\n        clearInterval(timer);\n      }\n      attempts++;\n      if (attempts > 100) {\n        console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n        clearInterval(timer);\n      }\n    }, 10, root)\n  }\n})(window);","application/vnd.bokehjs_exec.v0+json":""},"metadata":{"application/vnd.bokehjs_exec.v0+json":{"id":"1865"}},"output_type":"display_data"}],"source":["\n","\n","l = layout([\n","    [plot],\n","])\n","l.sizing_mode = \"scale_both\"\n","\n","\n","# show\n","output_file('Covid-19_Literature-Dashboard.html')\n","show(l)"]}],"metadata":{"colab":{"authorship_tag":"ABX9TyNFV+jVpC/z3P4CMfwl1iCv","collapsed_sections":[],"mount_file_id":"1xyfTqcpfg9nws9ObpIyDI0p47w5jTOjV","provenance":[],"toc_visible":true},"kernelspec":{"display_name":"Python 3.7.4 ('base')","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.7.4"},"vscode":{"interpreter":{"hash":"40d3a090f54c6569ab1632332b64b2c03c39dcf918b08424e98f38b5ae0af88f"}}},"nbformat":4,"nbformat_minor":0}
